diff --git a/feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go b/feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go
index 661b95fb..c39c152c 100644
--- a/feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go
+++ b/feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go
@@ -214,7 +214,7 @@ func (tc *testCase) configureDUT(t *testing.T) {
 	tc.configDstAggregateDUT(agg, &dutDst)
 	aggPath := d.Interface(tc.aggID)
 	fptest.LogYgot(t, tc.aggID, aggPath, agg)
-	aggPath.Replace(t, agg)
+	aggPath.Update(t, agg)
 
 	srcp := tc.dutPorts[0]
 	srci := &telemetry.Interface{Name: ygot.String(srcp.Name())}
@@ -377,7 +377,7 @@ func (tc *testCase) verifyMinLinks(t *testing.T) {
 		{
 			desc:      "MinLink - 1",
 			downCount: 2,
-			want:      telemetry.Interface_OperStatus_LOWER_LAYER_DOWN,
+			want:      opDown,
 		},
 	}
 
