diff --git a/feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go b/feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go
index 661b95fb..d14e1cc2 100644
--- a/feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go
+++ b/feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go
@@ -377,7 +377,8 @@ func (tc *testCase) verifyMinLinks(t *testing.T) {
 		{
 			desc:      "MinLink - 1",
 			downCount: 2,
-			want:      telemetry.Interface_OperStatus_LOWER_LAYER_DOWN,
+			// waiting for bug fix: CSCwd06230
+			want:      opDown,
 		},
 	}
 
