diff --git a/feature/system/gnmi/metadata/tests/annotation_test/annotation_test.go b/feature/system/gnmi/metadata/tests/annotation_test/annotation_test.go
index 3c032e3..801b38c 100644
--- a/feature/system/gnmi/metadata/tests/annotation_test/annotation_test.go
+++ b/feature/system/gnmi/metadata/tests/annotation_test/annotation_test.go
@@ -26,12 +26,14 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/openconfig/featureprofiles/internal/deviations"
 	"github.com/openconfig/featureprofiles/internal/fptest"
 	"github.com/openconfig/ondatra"
 	"github.com/openconfig/ygot/util"
 	"google.golang.org/protobuf/encoding/prototext"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/testing/protocmp"
+	"github.com/openconfig/ygot/ygot"
 )
 
 func TestMain(m *testing.M) {
@@ -100,6 +102,14 @@ func TestGNMIMetadataAnnotation(t *testing.T) {
 			t.Errorf("Cannot build a gNMI SetRequest from proto message: %v", err)
 		}
 
+		if *deviations.NoMetadataSingleUpdate {
+			t.Log("Appneding an update to the metadata due to NoMetadataSingleUpdate deviation")
+			// accompaniedPath and accompaniedUpdateVal can be any valid oc path and value
+			accompaniedPath := dut.Config().System().Hostname()
+			accompaniedUpdateVal := dut.Config().System().Hostname().Get(t)
+			gpbSetRequest.Update = append(gpbSetRequest.Update, buildGNMIUpdate(t,accompaniedPath,&accompaniedUpdateVal))
+		}
+
 		t.Log("gnmiClient Set metadata annotation")
 		if _, err = gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
 			t.Errorf("gnmi.Set unexpected error: %v", err)
@@ -217,3 +227,26 @@ func extractMetadataAnnotation(getResponse *gpb.GetResponse, m proto.Message) er
 	}
 	return nil
 }
+
+// buildGNMIUpdate builds a gnmi update for a given ygot path and value.
+func buildGNMIUpdate(t *testing.T, yPath ygot.PathStruct, val interface{}) *gpb.Update{
+	path, _, errs := ygot.ResolvePath(yPath)
+	if errs != nil {
+		t.Fatalf("Could not resolve the ygot path; %v", errs)
+	}
+	js, err := ygot.Marshal7951(val, ygot.JSONIndent("  "), &ygot.RFC7951JSONConfig{AppendModuleName: true, PreferShadowPath: true})
+	if err != nil {
+		t.Fatalf("Could not encode value into JSON format: %v", err)
+	}
+	return  &gpb.Update{
+		Path: path,
+		Val: &gpb.TypedValue{
+			Value: &gpb.TypedValue_JsonIetfVal{
+				JsonIetfVal: js,
+			},
+		},
+	}
+
+}
+
diff --git a/internal/deviations/deviations.go b/internal/deviations/deviations.go
index 6e990bb6..3cd27f76 100644
--- a/internal/deviations/deviations.go
+++ b/internal/deviations/deviations.go
@@ -111,4 +111,6 @@ var (
 
 	ExplicitInterfaceInDefaultVRF = flag.Bool("deviation_explicit_interface_in_default_vrf", false,
 		"Device requires explicit attachment of an interface or subinterface to the default network instance. OpenConfig expects an unattached interface or subinterface to be implicitly part of the default network instance. Fully-compliant devices should pass with and without this deviation.")
+
+	NoMetadataSingleUpdate = flag.Bool("deviation_nometadata_single_update", true, "Device requires metadata update to be accompanied by another update for an OC path.")
 )
