diff --git a/feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go b/feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go
index 480f4010..bf9d537e 100644
--- a/feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go
+++ b/feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go
@@ -152,15 +152,17 @@ func TestDirectBackupNexthopGroup(t *testing.T) {
 			desc: "Disable ATE port-2",
 			applyImpairmentFn: func() {
 				ateP2 := ate.Port(t, "port2")
-				dutP2 := dut.Port(t, "port2")
+				// dutP2 := dut.Port(t, "port2")
 				ate.Actions().NewSetPortState().WithPort(ateP2).WithEnabled(false).Send(t)
-				dut.Telemetry().Interface(dutP2.Name()).OperStatus().Await(t, time.Minute, telemetry.Interface_OperStatus_DOWN)
+				// CSCwc88672: 791, telemetry call for interface operational status returns flaky output
+				// dut.Telemetry().Interface(dutP2.Name()).OperStatus().Await(t, time.Minute, telemetry.Interface_OperStatus_DOWN)
 			},
 			removeImpairmentFn: func() {
 				ateP2 := ate.Port(t, "port2")
-				dutP2 := dut.Port(t, "port2")
+				// dutP2 := dut.Port(t, "port2")
 				ate.Actions().NewSetPortState().WithPort(ateP2).WithEnabled(true).Send(t)
-				dut.Telemetry().Interface(dutP2.Name()).OperStatus().Await(t, time.Minute, telemetry.Interface_OperStatus_UP)
+				// CSCwc88672: 791, telemetry call for interface operational status returns flaky output
+				// dut.Telemetry().Interface(dutP2.Name()).OperStatus().Await(t, time.Minute, telemetry.Interface_OperStatus_UP)
 			},
 		},
 		{
@@ -179,6 +181,7 @@ func TestDirectBackupNexthopGroup(t *testing.T) {
 	}
 	for _, tc := range cases {
 		t.Run(tc.desc, func(t *testing.T) {
+			time.Sleep(20 * time.Second)
 			t.Run("Validate Baseline AFT Telemetry", func(t *testing.T) {
 				tcArgs.validateAftTelemetry(t)
 			})
