diff --git a/feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go b/feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go
index 441cc56a..afd8eb67 100644
--- a/feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go
+++ b/feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go
@@ -231,7 +231,8 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 
 	vrf := &telemetry.NetworkInstance{
 		Name:    ygot.String(vrfName),
-		Enabled: ygot.Bool(true),
+		// Waiting for fix: CSCwd04983: [B4 FP: TE-15.1] add support for /network-instances/network-instance/enabled
+		//Enabled: ygot.Bool(true),
 		Type:    telemetry.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF,
 		EnabledAddressFamilies: []telemetry.E_Types_ADDRESS_FAMILY{
 			telemetry.Types_ADDRESS_FAMILY_IPV4,
