diff --git a/feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go b/feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go
index 23771c60..90c72aab 100644
--- a/feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go
+++ b/feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go
@@ -72,7 +72,8 @@ const (
 	plenIPv4                 = 30
 	plenIPv6                 = 126
 	bgpPort                  = 179
-	peerGrpName              = "BGP-PEER-GROUP"
+	peerv4GrpName            = "BGP-PEER-GROUP-V4"
+	peerv6GrpName            = "BGP-PEER-GROUP-V6"
 	ateDstCIDR               = "192.0.2.6/32"
 )
 
@@ -107,6 +108,15 @@ var (
 	}
 )
 
+func configureRoutePolicy(t *testing.T, dut *ondatra.DUTDevice, name string, pr oc.E_RoutingPolicy_PolicyResultType) {
+	d := &oc.Root{}
+	rp := d.GetOrCreateRoutingPolicy()
+	pd := rp.GetOrCreatePolicyDefinition(name)
+	st := pd.GetOrCreateStatement("id-1")
+	st.GetOrCreateActions().PolicyResult = pr
+	gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+}
+
 // configureDUT configures all the interfaces and network instance on the DUT.
 func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 	dc := gnmi.OC()
@@ -162,30 +172,58 @@ func bgpWithNbr(as uint32, nbrs []*bgpNeighbor) *oc.NetworkInstance_Protocol {
 
 	g := bgp.GetOrCreateGlobal()
 	g.As = ygot.Uint32(as)
+	g.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
+	g.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST).Enabled = ygot.Bool(true)
 	g.RouterId = ygot.String(dutDst.IPv4)
 	bgpgr := g.GetOrCreateGracefulRestart()
 	bgpgr.Enabled = ygot.Bool(true)
 	bgpgr.RestartTime = ygot.Uint16(grRestartTime)
 	bgpgr.StaleRoutesTime = ygot.Uint16(grStaleRouteTime)
 
-	pg := bgp.GetOrCreatePeerGroup(peerGrpName)
+	pg := bgp.GetOrCreatePeerGroup(peerv4GrpName)
 	pg.PeerAs = ygot.Uint32(ateAS)
-	pg.PeerGroupName = ygot.String(peerGrpName)
+	pg.PeerGroupName = ygot.String(peerv4GrpName)
+
+	pgv6 := bgp.GetOrCreatePeerGroup(peerv6GrpName)
+	pgv6.PeerAs = ygot.Uint32(ateAS)
+	pgv6.PeerGroupName = ygot.String(peerv6GrpName)
+
+	if *deviations.RoutePolicyUnderPeerGroup {
+		pg1af4 := pg.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+		pg1af4.Enabled = ygot.Bool(true)
+
+		pg1rpl4 := pg1af4.GetOrCreateApplyPolicy()
+		pg1rpl4.SetExportPolicy([]string{"ALLOW"})
+		pg1rpl4.SetImportPolicy([]string{"ALLOW"})
+
+		pg1af6 := pgv6.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+		pg1af6.Enabled = ygot.Bool(true)
+		pg1rpl6 := pg1af6.GetOrCreateApplyPolicy()
+		pg1rpl6.SetExportPolicy([]string{"ALLOW"})
+		pg1rpl6.SetImportPolicy([]string{"ALLOW"})
+
+	}
 
 	for _, nbr := range nbrs {
 		if nbr.isV4 {
 			nv4 := bgp.GetOrCreateNeighbor(nbr.neighborip)
-			nv4.PeerGroup = ygot.String(peerGrpName)
+			nv4.PeerGroup = ygot.String(peerv4GrpName)
 			nv4.GetOrCreateTimers().HoldTime = ygot.Uint16(180)
+			nv4.GetOrCreateTimers().KeepaliveInterval = ygot.Uint16(60)
 			nv4.PeerAs = ygot.Uint32(nbr.as)
 			nv4.Enabled = ygot.Bool(true)
-			nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
+			af := nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+			af.Enabled = ygot.Bool(true)
 		} else {
 			nv6 := bgp.GetOrCreateNeighbor(nbr.neighborip)
-			nv6.PeerGroup = ygot.String(peerGrpName)
+			nv6.PeerGroup = ygot.String(peerv6GrpName)
+			nv6.GetOrCreateTimers().HoldTime = ygot.Uint16(180)
+			nv6.GetOrCreateTimers().KeepaliveInterval = ygot.Uint16(60)
 			nv6.PeerAs = ygot.Uint32(nbr.as)
 			nv6.Enabled = ygot.Bool(true)
-			nv6.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST).Enabled = ygot.Bool(true)
+			nv6.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+			af := nv6.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+			af.Enabled = ygot.Bool(true)
 		}
 	}
 	return ni_proto
@@ -234,6 +272,23 @@ func checkBgpStatus(t *testing.T, dut *ondatra.DUTDevice) {
 	} else {
 		t.Errorf("Expected Graceful restart timer: got %v, want %v", grTimerVal, grRestartTime)
 	}
+
+	t.Log("Waiting for BGP v4 prefix to be installed")
+	got, found := gnmi.Watch(t, dut, nbrPath.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Prefixes().Installed().State(), 180*time.Second, func(val *ygnmi.Value[uint32]) bool {
+		prefixCount, ok := val.Val()
+		return ok && prefixCount == routeCount
+	}).Await(t)
+	if !found {
+		t.Errorf("Installed prefixes v4 mismatch: got %v, want %v", got, routeCount)
+	}
+	t.Log("Waiting for BGP v6 prefix to be installed")
+	got, found = gnmi.Watch(t, dut, nbrPathv6.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST).Prefixes().Installed().State(), 180*time.Second, func(val *ygnmi.Value[uint32]) bool {
+		prefixCount, ok := val.Val()
+		return ok && prefixCount == routeCount
+	}).Await(t)
+	if !found {
+		t.Errorf("Installed prefixes v6 mismatch: got %v, want %v", got, routeCount)
+	}
 }
 
 func configureATE(t *testing.T, ate *ondatra.ATEDevice) []*ondatra.Flow {
@@ -398,6 +453,9 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
 	t.Run("configureDut", func(t *testing.T) {
 		t.Log("Start DUT interface Config")
 		configureDUT(t, dut)
+		if *deviations.RoutePolicyUnderPeerGroup {
+			configureRoutePolicy(t, dut, "ALLOW", oc.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE)
+		}
 	})
 
 	// Configure BGP+Neighbors on the DUT
@@ -456,9 +514,15 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
 	nbrPath := statePath.Neighbor(ateDst.IPv4)
 	t.Run("VerifyBGPNOTEstablished", func(t *testing.T) {
 		t.Log("Waiting for BGP neighbor to go to CONNECT state after applying ACL DENY policy...")
+		var bgpState oc.E_Bgp_Neighbor_SessionState
+		if *deviations.BGPStateActiveACLDeny {
+			bgpState = oc.Bgp_Neighbor_SessionState_ACTIVE
+		} else {
+			bgpState = oc.Bgp_Neighbor_SessionState_CONNECT
+		}
 		_, ok := gnmi.Watch(t, dut, nbrPath.SessionState().State(), 2*time.Minute, func(val *ygnmi.Value[oc.E_Bgp_Neighbor_SessionState]) bool {
 			currState, ok := val.Val()
-			return ok && currState == oc.Bgp_Neighbor_SessionState_CONNECT
+			return ok && currState == bgpState
 		}).Await(t)
 		if !ok {
 			fptest.LogQuery(t, "BGP reported state", nbrPath.State(), gnmi.Get(t, dut, nbrPath.State()))
@@ -485,7 +549,7 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
 
 	t.Run("VerifyBGPEstablished", func(t *testing.T) {
 		t.Logf("Waiting for BGP neighbor to establish...")
-		_, ok := gnmi.Watch(t, dut, nbrPath.SessionState().State(), time.Minute, func(val *ygnmi.Value[oc.E_Bgp_Neighbor_SessionState]) bool {
+		_, ok := gnmi.Watch(t, dut, nbrPath.SessionState().State(), 2*time.Minute, func(val *ygnmi.Value[oc.E_Bgp_Neighbor_SessionState]) bool {
 			currState, ok := val.Val()
 			return ok && currState == oc.Bgp_Neighbor_SessionState_ESTABLISHED
 		}).Await(t)
@@ -503,4 +567,5 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
 		sendTraffic(t, ate, allFlows, trafficDuration)
 		verifyNoPacketLoss(t, ate, allFlows)
 	})
+
 }
diff --git a/internal/deviations/deviations.go b/internal/deviations/deviations.go
index ae549f80..1a782b7f 100644
--- a/internal/deviations/deviations.go
+++ b/internal/deviations/deviations.go
@@ -157,4 +157,7 @@ var (
 
 	ISISInstanceEnabledNotRequired = flag.Bool("deviation_isis_instance_enabled_not_required", false,
 		"Don't set isis instance enable flag on the device if value is true, Default value is false and instance enable flag is set")
+
+	BGPStateActiveACLDeny = flag.Bool("deviation_bgp_state_active_acl_deny", false,
+		"Device requires bgp state to be active after ACL deny policy")
 )
