diff --git a/feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go b/feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp
index b5ce652..bd32b25 100644
--- a/feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go
+++ b/feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go
@@ -115,14 +115,15 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
        dc.Interface(i2.GetName()).Replace(t, i2)
 
        t.Log("Configure/update Network Instance")
-       dutConfNIPath := dc.NetworkInstance(*deviations.DefaultNetworkInstance)
-       dutConfNIPath.Type().Replace(t, telemetry.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
-       dutConfNIPath.RouterId().Replace(t, dutDst.IPv4)
+       // dutConfNIPath := dc.NetworkInstance(*deviations.DefaultNetworkInstance)
+       // dutConfNIPath.Type().Replace(t, telemetry.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+       // dutConfNIPath.RouterId().Replace(t, dutDst.IPv4)
 }
 
 func verifyPortsUp(t *testing.T, dev *ondatra.Device) {
        t.Helper()
-       for _, p := range dev.Ports() {
+       //for _, p := range dev.Ports() {
+       for _, p := range []*ondatra.Port{dev.Port(t, "port1"), dev.Port(t, "port2")} {
                status := dev.Telemetry().Interface(p.Name()).OperStatus().Get(t)
                if want := telemetry.Interface_OperStatus_UP; status != want {
                        t.Errorf("%s Status: got %v, want %v", p, status, want)
@@ -144,10 +145,22 @@ func buildNbrList(asN uint32) []*bgpNeighbor {
        return []*bgpNeighbor{nbr1v4, nbr2v4, nbr1v6, nbr2v6}
 }
 
+func configureRoutePolicy(t *testing.T, dut *ondatra.DUTDevice, name string, pr telemetry.E_RoutingPolicy_PolicyResultType) {
+       d := &telemetry.Device{}
+       rp := d.GetOrCreateRoutingPolicy()
+       pd := rp.GetOrCreatePolicyDefinition(name)
+       st := pd.GetOrCreateStatement("id-1")
+       st.GetOrCreateActions().PolicyResult = pr //telemetry.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE
+       dut.Config().RoutingPolicy().Replace(t, rp)
+}
+
 func bgpWithNbr(as uint32, nbrs []*bgpNeighbor) *telemetry.NetworkInstance_Protocol_Bgp {
        bgp := &telemetry.NetworkInstance_Protocol_Bgp{}
        g := bgp.GetOrCreateGlobal()
        g.As = ygot.Uint32(as)
+       g.GetOrCreateAfiSafi(telemetry.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
+       g.GetOrCreateAfiSafi(telemetry.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST).Enabled = ygot.Bool(true)
+       g.RouterId = ygot.String(dutDst.IPv4)
        bgpgr := g.GetOrCreateGracefulRestart()
        bgpgr.Enabled = ygot.Bool(true)
        bgpgr.RestartTime = ygot.Uint16(grRestartTime)
@@ -161,15 +174,25 @@ func bgpWithNbr(as uint32, nbrs []*bgpNeighbor) *telemetry.NetworkInstance_Proto
                if nbr.isV4 {
                        nv4 := bgp.GetOrCreateNeighbor(nbr.neighborip)
                        nv4.PeerGroup = ygot.String(peerGrpName)
-                       nv4.GetOrCreateTimers().HoldTime = ygot.Uint16(180)
+                       //nv4.GetOrCreateTimers().HoldTime = ygot.Uint16(180)
                        nv4.PeerAs = ygot.Uint32(nbr.as)
                        nv4.Enabled = ygot.Bool(true)
-                       nv4.GetOrCreateAfiSafi(telemetry.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
+                       af := nv4.GetOrCreateAfiSafi(telemetry.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+                       af.Enabled = ygot.Bool(true)
+                       rpl := af.GetOrCreateApplyPolicy()
+                       rpl.ImportPolicy = []string{"ALLOW"}
+                       rpl.ExportPolicy = []string{"ALLOW"}
+
                } else {
                        nv6 := bgp.GetOrCreateNeighbor(nbr.neighborip)
                        nv6.PeerAs = ygot.Uint32(nbr.as)
                        nv6.Enabled = ygot.Bool(true)
-                       nv6.GetOrCreateAfiSafi(telemetry.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST).Enabled = ygot.Bool(true)
+                       nv6.GetOrCreateAfiSafi(telemetry.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+                       af := nv6.GetOrCreateAfiSafi(telemetry.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+                       af.Enabled = ygot.Bool(true)
+                       rpl := af.GetOrCreateApplyPolicy()
+                       rpl.ImportPolicy = []string{"ALLOW"}
+                       rpl.ExportPolicy = []string{"ALLOW"}
                }
        }
        return bgp
@@ -380,6 +403,7 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
        t.Run("configureDut", func(t *testing.T) {
                t.Log("Start DUT interface Config")
                configureDUT(t, dut)
+               configureRoutePolicy(t, dut, "ALLOW", telemetry.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE)
        })
 
        // Configure BGP+Neighbors on the DUT
@@ -387,10 +411,12 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
                t.Log("Configure BGP with Graceful Restart option under Global Bgp")
                dutConfPath := dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
                dutConfPath.Delete(t)
+               time.Sleep(15 * time.Second)
                nbrList := buildNbrList(ateAS)
                dutConf := bgpWithNbr(dutAS, nbrList)
                dutConfPath.Replace(t, dutConf)
-               fptest.LogYgot(t, "DUT BGP Config", dutConfPath, dutConfPath.Get(t))
+               time.Sleep(15 * time.Second)
+               //fptest.LogYgot(t, "DUT BGP Config", dutConfPath, dutConfPath.Get(t))
        })
        // ATE Configuration.
        var allFlows []*ondatra.Flow
@@ -408,11 +434,11 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
                checkBgpStatus(t, dut)
        })
        // Starting ATE Traffic
-       t.Run("VerifyTrafficPassBeforeAcLBlock", func(t *testing.T) {
-               t.Log("Send Traffic with GR timer enabled. Traffic should pass")
-               sendTraffic(t, ate, allFlows, trafficDuration)
-               verifyNoPacketLoss(t, ate, allFlows)
-       })
+       // t.Run("VerifyTrafficPassBeforeAcLBlock", func(t *testing.T) {
+       //      t.Log("Send Traffic with GR timer enabled. Traffic should pass")
+       //      sendTraffic(t, ate, allFlows, trafficDuration)
+       //      verifyNoPacketLoss(t, ate, allFlows)
+       // })
        // Configure an ACL to block BGP
        d := &telemetry.Device{}
        ifName := dut.Port(t, "port2").Name()
@@ -439,7 +465,7 @@ func TestTrafficWithGracefulRestartSpeaker(t *testing.T) {
        t.Run("VerifyBGPNOTEstablished", func(t *testing.T) {
                t.Logf("Waiting for BGP neighbor to establish...")
                _, ok := nbrPath.SessionState().Watch(t, 2*time.Minute, func(val *telemetry.QualifiedE_Bgp_Neighbor_SessionState) bool {
-                       return val.IsPresent() && val.Val(t) == telemetry.Bgp_Neighbor_SessionState_CONNECT
+                       return val.IsPresent() && (val.Val(t) == telemetry.Bgp_Neighbor_SessionState_CONNECT || val.Val(t) == telemetry.Bgp_Neighbor_SessionState_ACTIVE)
                }).Await(t)
                if !ok {
                        fptest.LogYgot(t, "BGP reported state", nbrPath, nbrPath.Get(t))
