diff --git a/feature/lldp/tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go b/feature/lldp/tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go
index 339bd0d8..6a42f1b6 100644
--- a/feature/lldp/tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go
+++ b/feature/lldp/tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go
@@ -19,6 +19,7 @@ import (
 	"time"
 
 	"github.com/openconfig/featureprofiles/internal/confirm"
+	"github.com/openconfig/featureprofiles/internal/deviations"
 	"github.com/openconfig/featureprofiles/internal/fptest"
 	"github.com/openconfig/ondatra"
 	"github.com/openconfig/ondatra/gnmi"
@@ -143,9 +144,17 @@ func verifyNodeTelemetry(t *testing.T, node, peer gnmi.DeviceOrOpts, nodePort, p
 				return true
 			}
 			gotLen = len(intf.Neighbor)
-			return gotLen == 0
+			if *deviations.LLDPInterfaceConfigOverrideGlobal {
+				return gotLen > 0
+			} else {
+				return gotLen == 0
+			}
 		}).Await(t); !ok {
-			t.Errorf("Number of neighbors got: %d, want: 0.", gotLen)
+			if *deviations.LLDPInterfaceConfigOverrideGlobal {
+				t.Errorf("Number of neighbors got: %d, want: non-zero.", gotLen)
+			} else {
+				t.Errorf("Number of neighbors got: %d, want: 0.", gotLen)
+			}
 		}
 		return
 	}
diff --git a/internal/deviations/deviations.go b/internal/deviations/deviations.go
index e8e169b8..5ce48a10 100644
--- a/internal/deviations/deviations.go
+++ b/internal/deviations/deviations.go
@@ -151,4 +151,7 @@ var (
 
 	ISISInstanceEnabledNotRequired = flag.Bool("deviation_isis_instance_enabled_not_required", false,
 		"Don't set isis instance enable flag on the device if value is true, Default value is false and instance enable flag is set")
+
+	LLDPInterfaceConfigOverrideGlobal = flag.Bool("deviation_lldp_interface_config_override_global", false,
+		"Set this flag for LLDP interface config to override the global config")
 )
