diff --git a/feature/lldp/tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go b/feature/lldp/tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go
index 339bd0d8..c2684449 100644
--- a/feature/lldp/tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go
+++ b/feature/lldp/tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go
@@ -143,9 +143,9 @@ func verifyNodeTelemetry(t *testing.T, node, peer gnmi.DeviceOrOpts, nodePort, p
 				return true
 			}
 			gotLen = len(intf.Neighbor)
-			return gotLen == 0
+			return gotLen > 0
 		}).Await(t); !ok {
-			t.Errorf("Number of neighbors got: %d, want: 0.", gotLen)
+			t.Errorf("Number of neighbors got: %d, want: non-zero.", gotLen)
 		}
 		return
 	}
