diff --git a/feature/gribi/ate_tests/get_rpc_test/get_rpc_test.go b/feature/gribi/ate_tests/get_rpc_test/get_rpc_test.go
index aa88a04e..4dfd488c 100644
--- a/feature/gribi/ate_tests/get_rpc_test/get_rpc_test.go
+++ b/feature/gribi/ate_tests/get_rpc_test/get_rpc_test.go
@@ -16,6 +16,7 @@ package get_rpc_test
 
 import (
 	"context"
+	"flag"
 	"testing"
 	"time"
 
@@ -82,6 +83,7 @@ var (
 
 var (
 	ateDstNetCIDR = []string{"198.51.100.0/26", "198.51.100.64/26", "198.51.100.128/26"}
+	protocolName  = flag.String("protocol_name", "STATIC", "Name of the protocol to configure static route")
 )
 
 const (
@@ -318,7 +320,7 @@ func testIPv4LeaderActive(ctx context.Context, t *testing.T, args *testArgs) {
 	// are returned, with no entry returned for 198.51.100.192/64.
 	dc := args.dut.Config()
 	ni := dc.NetworkInstance(*deviations.DefaultNetworkInstance).
-		Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, "STATIC")
+		Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *protocolName)
 	static := &telemetry.NetworkInstance_Protocol_Static{
 		Prefix: ygot.String(staticCIDR),
 	}
