diff --git a/feature/system/gnmi/get/tests/get_test.go b/feature/system/gnmi/get/tests/get_test.go
index dbc37ced..fa39cc7a
--- a/feature/system/gnmi/get/tests/get_test.go
+++ b/feature/system/gnmi/get/tests/get_test.go
@@ -66,6 +66,7 @@ func TestGNMIGet(t *testing.T) {
 			Path: []*gpb.Path{{
 				// empty path indicates the root.
 			}},
+			Type:     gpb.GetRequest_CONFIG,
 			Encoding: gpb.Encoding_JSON_IETF,
 		},
 		wantGetResponse: &gpb.GetResponse{
@@ -85,6 +86,7 @@ func TestGNMIGet(t *testing.T) {
 				Origin: "openconfig",
 			},
 			Path:     []*gpb.Path{{}},
+			Type:     gpb.GetRequest_CONFIG,
 			Encoding: gpb.Encoding_JSON_IETF,
 		},
 		wantGetResponse: &gpb.GetResponse{
