diff --git a/feature/gribi/ate_tests/gribigo_compliance_test/gribigo_compliance_test.go b/feature/gribi/ate_tests/gribigo_compliance_test/gribigo_compliance_test.go
index 69e51bd2..04e5e9ca 100644
--- a/feature/gribi/ate_tests/gribigo_compliance_test/gribigo_compliance_test.go
+++ b/feature/gribi/ate_tests/gribigo_compliance_test/gribigo_compliance_test.go
@@ -38,6 +38,8 @@ var (
 
 	nonDefaultNI = flag.String("non_default_ni", "non-default-vrf", "non-default network-instance name")
 
+	protocolName = flag.String("protocol_name", "default", "Name of the protocol to configure static route")
+
 	dutPort1 = attrs.Attributes{
 		Desc:    "dutPort1",
 		IPv4:    "192.0.2.0",
@@ -160,7 +162,7 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 	d := &telemetry.Device{}
 	ni := d.GetOrCreateNetworkInstance(*nonDefaultNI)
 	ni.Type = telemetry.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF
-	ni.GetOrCreateProtocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, "static")
+	ni.GetOrCreateProtocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *protocolName)
 	dut.Config().NetworkInstance(*nonDefaultNI).Replace(t, ni)
 
 	nip := dut.Config().NetworkInstance(*nonDefaultNI)
