diff --git a/feature/gribi/ate_tests/leader_failover_test/leader_failover_test.go b/feature/gribi/ate_tests/leader_failover_test/leader_failover_test.go
index 792bc658..989efe2e 100644
--- a/feature/gribi/ate_tests/leader_failover_test/leader_failover_test.go
+++ b/feature/gribi/ate_tests/leader_failover_test/leader_failover_test.go
@@ -252,7 +252,9 @@ func TestLeaderFailover(t *testing.T) {
 	t.Run("SINGLE_PRIMARY/PERSISTENCE=DELETE", func(t *testing.T) {
 		// This is an indicator test for gRIBI persistence DELETE, so we
 		// do not skip based on *deviations.GRIBIPreserveOnly.
-
+		if *deviations.GRIBIPreserveOnly {
+			t.Skip()
+		}
 		// Set parameters for gRIBI client clientA.
 		// Set Persistence to false.
 		clientA := &gribi.Client{
@@ -291,7 +293,9 @@ func TestLeaderFailover(t *testing.T) {
 	t.Run("ShouldDelete", func(t *testing.T) {
 		// This is an indicator test for gRIBI persistence DELETE, so we
 		// do not skip based on *deviations.GRIBIPreserveOnly.
-
+		if *deviations.GRIBIPreserveOnly {
+			t.Skip()
+		}
 		t.Logf("Verify through Telemetry and Traffic that the route to %s has been deleted after gRIBI client disconnected", ateDstNetCIDR)
 
 		t.Run("VerifyNoAFT", func(t *testing.T) {
