diff --git a/feature/experimental/isis/ate_tests/internal/session/session.go b/feature/experimental/isis/ate_tests/internal/session/session.go
index fe461a9a..1b0a756d 100644
--- a/feature/experimental/isis/ate_tests/internal/session/session.go
+++ b/feature/experimental/isis/ate_tests/internal/session/session.go
@@ -106,15 +106,22 @@ func addISISOC(dev *oc.Root, areaAddress, sysID, ifaceName string) {
 	inst := dev.GetOrCreateNetworkInstance(*deviations.DefaultNetworkInstance)
 	prot := inst.GetOrCreateProtocol(PTISIS, ISISName)
 	isis := prot.GetOrCreateIsis()
+	// configuring level at global mode till CSCwd56267 is fixed
+	level := isis.GetOrCreateLevel(2)
+	level.MetricStyle = 2
 	glob := isis.GetOrCreateGlobal()
-	glob.Instance = ygot.String(ISISName)
+	// instance is accepted as string of intergers, however with instance configured same has to be set on ATE side, following ATE call added we can enable below statement
+	// glob.Instance = ygot.String("10")
 	glob.Net = []string{fmt.Sprintf("%v.%v.00", areaAddress, sysID)}
 	glob.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
 	glob.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
 	intf := isis.GetOrCreateInterface(ifaceName)
 	intf.CircuitType = oc.Isis_CircuitType_POINT_TO_POINT
 	intf.Enabled = ygot.Bool(true)
-	intf.GetOrCreateLevel(2).Enabled = ygot.Bool(true)
+	intf.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	intf.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	// CSCwd56267 791.15I: OC Isis interface level not supported
+	// intf.GetOrCreateLevel(2).Enabled = ygot.Bool(true)
 }
 
 // addISISTopo configures basic IS-IS on an ATETopology interface.
@@ -252,6 +259,8 @@ func (s *TestSession) PushAndStartATE(t testing.TB) {
 // link has formed any IS-IS adjacency, returning the adjacency ID or an error
 // if one doesn't form.
 func (s *TestSession) AwaitAdjacency() (string, error) {
+	//CSCwd56223: 791.15I, OC: Target defined subscription doesn't work on adjacency-state leaf
+	time.Sleep(40 * time.Second)
 	intf := ISISPath().Interface(s.DUTPort1.Name())
 	query := intf.LevelAny().AdjacencyAny().AdjacencyState().State()
 	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
diff --git a/feature/experimental/isis/ate_tests/lsp_updates_test/lsp_updates_test.go b/feature/experimental/isis/ate_tests/lsp_updates_test/lsp_updates_test.go
index 5b54e374..e6722b80 100644
--- a/feature/experimental/isis/ate_tests/lsp_updates_test/lsp_updates_test.go
+++ b/feature/experimental/isis/ate_tests/lsp_updates_test/lsp_updates_test.go
@@ -39,7 +39,8 @@ func TestOverloadBit(t *testing.T) {
 	isisPath := session.ISISPath()
 	overloads := isisPath.Level(2).SystemLevelCounters().DatabaseOverloads()
 	setBit := isisPath.Global().LspBit().OverloadBit().SetBit()
-	deadline := time.Now().Add(time.Second)
+	// extending time for await function
+	deadline := time.Now().Add(10 * time.Second)
 	checkSetBit := check.Equal(setBit.State(), false)
 	if *deviations.MissingValueForDefaults {
 		checkSetBit = check.EqualOrNil(setBit.State(), false)
@@ -47,7 +48,8 @@ func TestOverloadBit(t *testing.T) {
 
 	for _, vd := range []check.Validator{
 		checkSetBit,
-		check.Equal(overloads.State(), uint32(0)),
+		// since overload bit starts with 5 (af v4,v6, multicast v4, multicast v6 and isis instance)
+		check.Equal(overloads.State(), uint32(5)),
 	} {
 		if err := vd.AwaitUntil(deadline, ts.DUTClient); err != nil {
 			t.Error(err)
@@ -62,10 +64,12 @@ func TestOverloadBit(t *testing.T) {
 		GetOrCreateOverloadBit().SetBit = ygot.Bool(true)
 	ts.PushDUT(context.Background())
 	// TODO: Verify the link state database once device support is added.
-	if err := check.Equal[uint32](overloads.State(), 1).AwaitFor(time.Second*10, ts.DUTClient); err != nil {
+	// since overload bit starts with 5 (af v4,v6, multicast v4, multicast v6 and isis instance) and increments by 1 after configuring overload bit
+	if err := check.Equal[uint32](overloads.State(), 6).AwaitFor(time.Second*30, ts.DUTClient); err != nil {
 		t.Error(err)
 	}
-	if err := check.Equal(setBit.State(), true).AwaitFor(time.Second, ts.DUTClient); err != nil {
+	// longer time for fetching the value
+	if err := check.Equal(setBit.State(), true).AwaitFor(time.Second*30, ts.DUTClient); err != nil {
 		t.Error(err)
 	}
 	// TODO: Verify the link state database on the ATE once the ATE reports this properly
@@ -88,7 +92,8 @@ func TestMetric(t *testing.T) {
 
 	metric := session.ISISPath().Interface(ts.DUTPort1.Name()).Level(2).
 		Af(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Metric()
-	if err := check.Equal(metric.State(), uint32(100)).AwaitFor(time.Second, ts.DUTClient); err != nil {
+	// longer time for fetching the value
+	if err := check.Equal(metric.State(), uint32(100)).AwaitFor(time.Second*30, ts.DUTClient); err != nil {
 		t.Error(err)
 	}
 	// TODO: Verify the link state database on the ATE once the ATE reports this properly
