diff --git a/feature/experimental/qos/ate_tests/mixed_sp_wrr_traffic_test/mixed_sp_wrr_traffic_test.go b/feature/experimental/qos/ate_tests/mixed_sp_wrr_traffic_test/mixed_sp_wrr_traffic_test.go
index 51c6dd7..123f0d1 100644
--- a/feature/experimental/qos/ate_tests/mixed_sp_wrr_traffic_test/mixed_sp_wrr_traffic_test.go
+++ b/feature/experimental/qos/ate_tests/mixed_sp_wrr_traffic_test/mixed_sp_wrr_traffic_test.go
@@ -116,13 +116,13 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			"BE0": dp3.Name() + "-0",
 		},
 		ondatra.CISCO: {
-			"NC1": "7",
-			"AF4": "4",
-			"AF3": "3",
-			"AF2": "2",
-			"AF1": "0",
-			"BE1": "1",
-			"BE0": "1",
+			"NC1": "tc7",
+			"AF4": "tc6",
+			"AF3": "tc5",
+			"AF2": "tc4",
+			"AF1": "tc3",
+			"BE1": "tc1",
+			"BE0": "tc2",
 		},
 		ondatra.NOKIA: {
 			"NC1": "7",
@@ -178,7 +178,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			queue:                 queueMap[dut.Vendor()]["AF1"],
 			inputIntf:             intf1,
 		},
-		"intf1-be1": {
+		"intf1-be0": {
 			frameSize:             1000,
 			trafficRate:           2,
 			expectedThroughputPct: 100.0,
@@ -186,7 +186,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			queue:                 queueMap[dut.Vendor()]["BE0"],
 			inputIntf:             intf1,
 		},
-		"intf1-be0": {
+		"intf1-be1": {
 			frameSize:             1000,
 			trafficRate:           0.5,
 			dscp:                  0,
@@ -234,7 +234,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			queue:                 queueMap[dut.Vendor()]["AF1"],
 			inputIntf:             intf2,
 		},
-		"intf2-be1": {
+		"intf2-be0": {
 			frameSize:             1000,
 			trafficRate:           2,
 			dscp:                  4,
@@ -242,7 +242,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			queue:                 queueMap[dut.Vendor()]["BE0"],
 			inputIntf:             intf2,
 		},
-		"intf2-be0": {
+		"intf2-be1": {
 			frameSize:             1000,
 			trafficRate:           0.5,
 			expectedThroughputPct: 100.0,
@@ -296,7 +296,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			queue:                 queueMap[dut.Vendor()]["AF1"],
 			inputIntf:             intf1,
 		},
-		"intf1-be1": {
+		"intf1-be0": {
 			frameSize:             1000,
 			trafficRate:           1,
 			expectedThroughputPct: 0.0,
@@ -304,7 +304,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			queue:                 queueMap[dut.Vendor()]["BE0"],
 			inputIntf:             intf1,
 		},
-		"intf1-be0": {
+		"intf1-be1": {
 			frameSize:             1000,
 			trafficRate:           1,
 			dscp:                  0,
@@ -352,7 +352,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			queue:                 queueMap[dut.Vendor()]["AF1"],
 			inputIntf:             intf2,
 		},
-		"intf2-be1": {
+		"intf2-be0": {
 			frameSize:             1000,
 			trafficRate:           7,
 			dscp:                  4,
@@ -360,7 +360,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			queue:                 queueMap[dut.Vendor()]["BE0"],
 			inputIntf:             intf2,
 		},
-		"intf2-be0": {
+		"intf2-be1": {
 			frameSize:             1000,
 			trafficRate:           1,
 			expectedThroughputPct: 0.0,
@@ -392,7 +392,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 		},
 		"intf1-af3": {
 			frameSize:             1000,
-			trafficRate:           40,
+			trafficRate:           32,
 			expectedThroughputPct: 50.0,
 			dscp:                  24,
 			queue:                 queueMap[dut.Vendor()]["AF3"],
@@ -400,7 +400,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 		},
 		"intf1-af2": {
 			frameSize:             1000,
-			trafficRate:           8,
+			trafficRate:           16,
 			expectedThroughputPct: 50.0,
 			dscp:                  16,
 			queue:                 queueMap[dut.Vendor()]["AF2"],
@@ -408,20 +408,20 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 		},
 		"intf1-af1": {
 			frameSize:             1000,
-			trafficRate:           12,
+			trafficRate:           8,
 			expectedThroughputPct: 50.0,
 			dscp:                  8, queue: queueMap[dut.Vendor()]["AF1"],
 			inputIntf: intf1,
 		},
-		"intf1-be1": {
+		"intf1-be0": {
 			frameSize:             1000,
-			trafficRate:           1,
+			trafficRate:           4,
 			expectedThroughputPct: 50.0,
 			dscp:                  4,
 			queue:                 queueMap[dut.Vendor()]["BE0"],
 			inputIntf:             intf1,
 		},
-		"intf1-be0": {
+		"intf1-be1": {
 			frameSize:             1000,
 			trafficRate:           1,
 			dscp:                  0,
@@ -447,7 +447,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 		},
 		"intf2-af3": {
 			frameSize:             1000,
-			trafficRate:           24,
+			trafficRate:           32,
 			expectedThroughputPct: 50.0,
 			dscp:                  24,
 			queue:                 queueMap[dut.Vendor()]["AF3"],
@@ -455,7 +455,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 		},
 		"intf2-af2": {
 			frameSize:             1000,
-			trafficRate:           24,
+			trafficRate:           16,
 			expectedThroughputPct: 50.0,
 			dscp:                  16,
 			queue:                 queueMap[dut.Vendor()]["AF2"],
@@ -463,21 +463,21 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 		},
 		"intf2-af1": {
 			frameSize:             1000,
-			trafficRate:           4,
+			trafficRate:           8,
 			expectedThroughputPct: 50.0,
 			dscp:                  8,
 			queue:                 queueMap[dut.Vendor()]["AF1"],
 			inputIntf:             intf2,
 		},
-		"intf2-be1": {
+		"intf2-be0": {
 			frameSize:             1000,
-			trafficRate:           7,
+			trafficRate:           4,
 			dscp:                  4,
 			expectedThroughputPct: 50.0,
 			queue:                 queueMap[dut.Vendor()]["BE0"],
 			inputIntf:             intf2,
 		},
-		"intf2-be0": {
+		"intf2-be1": {
 			frameSize:             1000,
 			trafficRate:           1,
 			expectedThroughputPct: 50.0,
@@ -544,7 +544,7 @@ func TestMixedSPWrrTraffic(t *testing.T) {
 			t.Logf("Running traffic 2 on DUT interfaces: %s => %s ", dp2.Name(), dp3.Name())
 			t.Logf("Sending traffic flows: \n%v\n\n", trafficFlows)
 			ate.Traffic().Start(t, flows...)
-			time.Sleep(120 * time.Second)
+			time.Sleep(10 * time.Second)
 			ate.Traffic().Stop(t)
 			time.Sleep(30 * time.Second)
 
@@ -649,105 +649,111 @@ func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
 		dscpSet      []uint8
 	}{{
 		desc:         "classifier_ipv4_be1",
-		name:         "dscp_based_classifier_ipv4",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV4,
 		termID:       "0",
-		targetGrpoup: "target-group-BE1",
+		targetGrpoup: "tc1",
 		dscpSet:      []uint8{0, 1, 2, 3},
 	}, {
 		desc:         "classifier_ipv4_be0",
-		name:         "dscp_based_classifier_ipv4",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV4,
 		termID:       "1",
-		targetGrpoup: "target-group-BE0",
+		targetGrpoup: "tc2",
 		dscpSet:      []uint8{4, 5, 6, 7},
 	}, {
 		desc:         "classifier_ipv4_af1",
-		name:         "dscp_based_classifier_ipv4",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV4,
 		termID:       "2",
-		targetGrpoup: "target-group-AF1",
+		targetGrpoup: "tc3",
 		dscpSet:      []uint8{8, 9, 10, 11},
 	}, {
 		desc:         "classifier_ipv4_af2",
-		name:         "dscp_based_classifier_ipv4",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV4,
 		termID:       "3",
-		targetGrpoup: "target-group-AF2",
+		targetGrpoup: "tc4",
 		dscpSet:      []uint8{16, 17, 18, 19},
 	}, {
 		desc:         "classifier_ipv4_af3",
-		name:         "dscp_based_classifier_ipv4",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV4,
 		termID:       "4",
-		targetGrpoup: "target-group-AF3",
+		targetGrpoup: "tc5",
 		dscpSet:      []uint8{24, 25, 26, 27},
 	}, {
 		desc:         "classifier_ipv4_af4",
-		name:         "dscp_based_classifier_ipv4",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV4,
 		termID:       "5",
-		targetGrpoup: "target-group-AF4",
+		targetGrpoup: "tc6",
 		dscpSet:      []uint8{32, 33, 34, 35},
 	}, {
 		desc:         "classifier_ipv4_nc1",
-		name:         "dscp_based_classifier_ipv4",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV4,
 		termID:       "6",
-		targetGrpoup: "target-group-NC1",
+		targetGrpoup: "tc7",
 		dscpSet:      []uint8{48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59},
 	}, {
 		desc:         "classifier_ipv6_be1",
-		name:         "dscp_based_classifier_ipv6",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV6,
-		termID:       "0",
-		targetGrpoup: "target-group-BE1",
+		termID:       "7",
+		targetGrpoup: "tc1",
 		dscpSet:      []uint8{0, 1, 2, 3},
 	}, {
 		desc:         "classifier_ipv6_be0",
-		name:         "dscp_based_classifier_ipv6",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV6,
-		termID:       "1",
-		targetGrpoup: "target-group-BE0",
+		termID:       "8",
+		targetGrpoup: "tc2",
 		dscpSet:      []uint8{4, 5, 6, 7},
 	}, {
 		desc:         "classifier_ipv6_af1",
-		name:         "dscp_based_classifier_ipv6",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV6,
-		termID:       "2",
-		targetGrpoup: "target-group-AF1",
+		termID:       "9",
+		targetGrpoup: "tc3",
 		dscpSet:      []uint8{8, 9, 10, 11},
 	}, {
 		desc:         "classifier_ipv6_af2",
-		name:         "dscp_based_classifier_ipv6",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV6,
-		termID:       "3",
-		targetGrpoup: "target-group-AF2",
+		termID:       "10",
+		targetGrpoup: "tc4",
 		dscpSet:      []uint8{16, 17, 18, 19},
 	}, {
 		desc:         "classifier_ipv6_af3",
-		name:         "dscp_based_classifier_ipv6",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV6,
-		termID:       "4",
-		targetGrpoup: "target-group-AF3",
+		termID:       "12",
+		targetGrpoup: "tc6",
 		dscpSet:      []uint8{24, 25, 26, 27},
 	}, {
 		desc:         "classifier_ipv6_af4",
-		name:         "dscp_based_classifier_ipv6",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV6,
-		termID:       "5",
-		targetGrpoup: "target-group-AF4",
+		termID:       "13",
+		targetGrpoup: "tc6",
 		dscpSet:      []uint8{32, 33, 34, 35},
 	}, {
 		desc:         "classifier_ipv6_nc1",
-		name:         "dscp_based_classifier_ipv6",
+		name:         "dscp_based_classifier",
 		classType:    oc.Qos_Classifier_Type_IPV6,
-		termID:       "6",
-		targetGrpoup: "target-group-NC1",
+		termID:       "14",
+		targetGrpoup: "tc7",
 		dscpSet:      []uint8{48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59},
 	}}
 
 	t.Logf("qos Classifiers config: %v", classifiers)
+	queues := []string{"tc7", "tc6", "tc5", "tc4", "tc3", "tc2", "tc1"}
+	for _, queue := range queues {
+		q1 := q.GetOrCreateQueue(queue)
+		q1.Name = ygot.String(queue)
+		gnmi.Update(t, dut, gnmi.OC().Qos().Queue(*q1.Name).Config(), q1)
+	}
 	for _, tc := range classifiers {
 		classifier := q.GetOrCreateClassifier(tc.name)
 		classifier.SetName(tc.name)
@@ -760,14 +766,19 @@ func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
 		term.SetId(tc.termID)
 		action := term.GetOrCreateActions()
 		action.SetTargetGroup(tc.targetGrpoup)
+
 		condition := term.GetOrCreateConditions()
-		if tc.name == "dscp_based_classifier_ipv4" {
+		if tc.classType == oc.Qos_Classifier_Type_IPV4 {
 			condition.GetOrCreateIpv4().SetDscpSet(tc.dscpSet)
-		} else if tc.name == "dscp_based_classifier_ipv6" {
+		} else {
 			condition.GetOrCreateIpv6().SetDscpSet(tc.dscpSet)
+
 		}
-		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+		fwdgroups := q.GetOrCreateForwardingGroup(tc.targetGrpoup)
+		fwdgroups.Name = ygot.String(tc.targetGrpoup)
+		fwdgroups.OutputQueue = ygot.String(tc.targetGrpoup)
 	}
+	gnmi.Update(t, dut, gnmi.OC().Qos().Config(), q)
 
 	t.Logf("Create qos input classifier config")
 	classifierIntfs := []struct {
@@ -799,57 +810,16 @@ func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
 
 	t.Logf("qos input classifier config: %v", classifierIntfs)
 	for _, tc := range classifierIntfs {
+		d = &oc.Root{}
+		q = d.GetOrCreateQos()
 		i := q.GetOrCreateInterface(tc.intf)
-		i.SetInterfaceId(tc.intf)
-		c := i.GetOrCreateInput().GetOrCreateClassifier(tc.inputClassifierType)
-		c.SetType(tc.inputClassifierType)
-		c.SetName(tc.classifier)
-		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
-	}
+		i.InterfaceId = ygot.String(tc.intf)
+		c := i.GetOrCreateInput()
+		c.GetOrCreateClassifier(oc.Input_Classifier_Type_IPV4).Name = ygot.String("dscp_based_classifier")
+		c.GetOrCreateClassifier(oc.Input_Classifier_Type_IPV6).Name = ygot.String("dscp_based_classifier")
+		c.GetOrCreateClassifier(oc.Input_Classifier_Type_MPLS).Name = ygot.String("dscp_based_classifier")
 
-	t.Logf("Create qos forwarding groups config")
-	forwardingGroups := []struct {
-		desc         string
-		queueName    string
-		targetGrpoup string
-	}{{
-		desc:         "forwarding-group-BE1",
-		queueName:    "BE1",
-		targetGrpoup: "target-group-BE1",
-	}, {
-		desc:         "forwarding-group-BE0",
-		queueName:    "BE0",
-		targetGrpoup: "target-group-BE0",
-	}, {
-		desc:         "forwarding-group-AF1",
-		queueName:    "AF1",
-		targetGrpoup: "target-group-AF1",
-	}, {
-		desc:         "forwarding-group-AF2",
-		queueName:    "AF2",
-		targetGrpoup: "target-group-AF2",
-	}, {
-		desc:         "forwarding-group-AF3",
-		queueName:    "AF3",
-		targetGrpoup: "target-group-AF3",
-	}, {
-		desc:         "forwarding-group-AF4",
-		queueName:    "AF4",
-		targetGrpoup: "target-group-AF4",
-	}, {
-		desc:         "forwarding-group-NC1",
-		queueName:    "NC1",
-		targetGrpoup: "target-group-NC1",
-	}}
-
-	t.Logf("qos forwarding groups config: %v", forwardingGroups)
-	for _, tc := range forwardingGroups {
-		fwdGroup := q.GetOrCreateForwardingGroup(tc.targetGrpoup)
-		fwdGroup.SetName(tc.targetGrpoup)
-		fwdGroup.SetOutputQueue(tc.queueName)
-		queue := q.GetOrCreateQueue(tc.queueName)
-		queue.SetName(tc.queueName)
-		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+		gnmi.Update(t, dut, gnmi.OC().Qos().Config(), q)
 	}
 
 	t.Logf("Create qos scheduler policies config")
@@ -866,65 +836,65 @@ func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
 		desc:         "scheduler-policy-BE1",
 		sequence:     uint32(1),
 		priority:     oc.Scheduler_Priority_UNSET,
-		inputID:      "BE1",
+		inputID:      "tc2",
 		inputType:    oc.Input_InputType_QUEUE,
-		weight:       uint64(1),
-		queueName:    "BE1",
-		targetGrpoup: "target-group-BE1",
+		weight:       uint64(4),
+		queueName:    "tc2",
+		targetGrpoup: "tc2",
 	}, {
 		desc:         "scheduler-policy-BE0",
 		sequence:     uint32(1),
 		priority:     oc.Scheduler_Priority_UNSET,
-		inputID:      "BE0",
+		inputID:      "tc1",
 		inputType:    oc.Input_InputType_QUEUE,
-		weight:       uint64(4),
-		queueName:    "BE0",
-		targetGrpoup: "target-group-BE0",
+		weight:       uint64(1),
+		queueName:    "tc1",
+		targetGrpoup: "tc1",
 	}, {
 		desc:         "scheduler-policy-AF1",
 		sequence:     uint32(1),
 		priority:     oc.Scheduler_Priority_UNSET,
-		inputID:      "AF1",
+		inputID:      "tc3",
 		inputType:    oc.Input_InputType_QUEUE,
 		weight:       uint64(8),
-		queueName:    "AF1",
-		targetGrpoup: "target-group-AF1",
+		queueName:    "tc3",
+		targetGrpoup: "tc3",
 	}, {
 		desc:         "scheduler-policy-AF2",
 		sequence:     uint32(1),
 		priority:     oc.Scheduler_Priority_UNSET,
-		inputID:      "AF2",
+		inputID:      "tc4",
 		inputType:    oc.Input_InputType_QUEUE,
 		weight:       uint64(16),
-		queueName:    "AF2",
-		targetGrpoup: "target-group-AF2",
+		queueName:    "tc4",
+		targetGrpoup: "tc4",
 	}, {
 		desc:         "scheduler-policy-AF3",
 		sequence:     uint32(1),
 		priority:     oc.Scheduler_Priority_UNSET,
-		inputID:      "AF3",
+		inputID:      "tc5",
 		inputType:    oc.Input_InputType_QUEUE,
 		weight:       uint64(32),
-		queueName:    "AF3",
-		targetGrpoup: "target-group-AF3",
+		queueName:    "tc5",
+		targetGrpoup: "tc5",
 	}, {
 		desc:         "scheduler-policy-AF4",
-		sequence:     uint32(0),
+		sequence:     uint32(2),
 		priority:     oc.Scheduler_Priority_STRICT,
-		inputID:      "AF4",
+		inputID:      "tc6",
 		inputType:    oc.Input_InputType_QUEUE,
-		weight:       uint64(100),
-		queueName:    "AF4",
-		targetGrpoup: "target-group-AF4",
+		weight:       uint64(6),
+		queueName:    "tc6",
+		targetGrpoup: "tc6",
 	}, {
 		desc:         "scheduler-policy-NC1",
-		sequence:     uint32(0),
+		sequence:     uint32(2),
 		priority:     oc.Scheduler_Priority_STRICT,
-		inputID:      "NC1",
+		inputID:      "tc7",
 		inputType:    oc.Input_InputType_QUEUE,
-		weight:       uint64(200),
-		queueName:    "NC1",
-		targetGrpoup: "target-group-NC1",
+		weight:       uint64(7),
+		queueName:    "tc7",
+		targetGrpoup: "tc7",
 	}}
 
 	schedulerPolicy := q.GetOrCreateSchedulerPolicy("scheduler")
@@ -936,11 +906,12 @@ func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
 		s.SetPriority(tc.priority)
 		input := s.GetOrCreateInput(tc.inputID)
 		input.SetId(tc.inputID)
-		input.SetInputType(tc.inputType)
+		//input.SetInputType(tc.inputType)
 		input.SetQueue(tc.queueName)
 		input.SetWeight(tc.weight)
-		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+
 	}
+	gnmi.Update(t, dut, gnmi.OC().Qos().Config(), q)
 
 	t.Logf("Create qos output interface config")
 	schedulerIntfs := []struct {
@@ -949,31 +920,31 @@ func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
 		scheduler string
 	}{{
 		desc:      "output-interface-BE1",
-		queueName: "BE1",
+		queueName: "tc1",
 		scheduler: "scheduler",
 	}, {
 		desc:      "output-interface-BE0",
-		queueName: "BE0",
+		queueName: "tc2",
 		scheduler: "scheduler",
 	}, {
 		desc:      "output-interface-AF1",
-		queueName: "AF1",
+		queueName: "tc3",
 		scheduler: "scheduler",
 	}, {
 		desc:      "output-interface-AF2",
-		queueName: "AF2",
+		queueName: "tc4",
 		scheduler: "scheduler",
 	}, {
 		desc:      "output-interface-AF3",
-		queueName: "AF3",
+		queueName: "tc5",
 		scheduler: "scheduler",
 	}, {
 		desc:      "output-interface-AF4",
-		queueName: "AF4",
+		queueName: "tc6",
 		scheduler: "scheduler",
 	}, {
 		desc:      "output-interface-NC1",
-		queueName: "NC1",
+		queueName: "tc7",
 		scheduler: "scheduler",
 	}}
 
@@ -986,6 +957,6 @@ func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
 		schedulerPolicy.SetName(tc.scheduler)
 		queue := output.GetOrCreateQueue(tc.queueName)
 		queue.SetName(tc.queueName)
-		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+		gnmi.Update(t, dut, gnmi.OC().Qos().Config(), q)
 	}
 }
