diff --git a/feature/gribi/ate_tests/ordering_ack_test/ordering_ack_test.go b/feature/gribi/ate_tests/ordering_ack_test/ordering_ack_test.go
index 7c14d0c..929a824 100644
--- a/feature/gribi/ate_tests/ordering_ack_test/ordering_ack_test.go
+++ b/feature/gribi/ate_tests/ordering_ack_test/ordering_ack_test.go
@@ -232,9 +232,13 @@ func testModifyNHG(t *testing.T, args *testArgs) {
 	t.Run("Telemetry", func(t *testing.T) {
 		got := aftNextHopWeights(t, args.dut, nhgIndex, *deviations.DefaultNetworkInstance)
 		want := []uint64{nhWeight}
-		ok := cmp.Equal(want, got, cmpopts.SortSlices(func(a, b uint64) bool { return a < b }))
-		if !ok {
-			t.Errorf("next-hop-group/next-hop/state/weight got %v, want %v", got, want)
+		// when a next hop group (nhg) has only one next hop, most FIB implemenation map the nhg to a single path and ignore the weight.
+		// In this case, AFT may returns no value or zero as weight, so validate weights only for nhg with more than on nh.
+		if len(want) > 1 {
+			ok := cmp.Equal(want, got, cmpopts.SortSlices(func(a, b uint64) bool { return a < b }))
+			if !ok {
+				t.Errorf("next-hop-group/next-hop/state/weight got %v, want %v", got, want)
+			}
 		}
 	})
 }
@@ -320,11 +324,14 @@ func testModifyNHGIPv4(t *testing.T, args *testArgs) {
 	t.Run("Telemetry", func(t *testing.T) {
 		got := aftNextHopWeights(t, args.dut, nhgIndex, *deviations.DefaultNetworkInstance)
 		want := []uint64{nhWeight}
-		ok := cmp.Equal(want, got, cmpopts.SortSlices(func(a, b uint64) bool { return a < b }))
-		if !ok {
-			t.Errorf("next-hop-group/next-hop/state/weight got %v, want %v", got, want)
+		// if a next hop group (nhg) has only one next hop, most FIB implemenation map the nhg to a single path and ignore the weight.
+		// In this case, AFT may returns no value or zero as weight, so validate weights only for nhg with more than on nh.
+		if len(want) > 1 {
+			ok := cmp.Equal(want, got, cmpopts.SortSlices(func(a, b uint64) bool { return a < b }))
+			if !ok {
+				t.Errorf("next-hop-group/next-hop/state/weight got %v, want %v", got, want)
+			}
 		}
-
 		ipv4Path := args.dut.Telemetry().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().Ipv4Entry(ateDstNetCIDR)
 		if got, want := ipv4Path.Prefix().Get(t), ateDstNetCIDR; got != want {
 			t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
@@ -348,6 +355,7 @@ func aftNextHopWeights(t *testing.T, dut *ondatra.DUTDevice, nhg uint64, network
 		}
 	}
 	if nhgD == nil {
+		t.Errorf("next-hop-group with programing id %d is not found in AFT response", nhg)
 		return []uint64{}
 	}
 
