--- a/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go
+++ b/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go
@@ -90,11 +90,13 @@ func TestStandbyControllerCardReboot(t *testing.T) {
        t.Logf("Detected rpStandby: %v, rpActive: %v", rpStandby, rpActive)

        gnoiClient := dut.RawAPIs().GNOI().Default(t)
+       keys := map[string]string{"name": rpStandby}
        rebootSubComponentRequest := &spb.RebootRequest{
                Method: spb.RebootMethod_COLD,
                Subcomponents: []*tpb.Path{
                        {
-                               Elem: []*tpb.PathElem{{Name: rpStandby}},
+                               Origin: "openconfig",
+                               Elem:   []*tpb.PathElem{{Name: "components"}, {Name: "component", Key: keys}, {Name: "state"}, {Name: "location"}},
                        },
                },
        }
@@ -147,11 +149,14 @@ func TestLinecardReboot(t *testing.T) {
        }

        gnoiClient := dut.RawAPIs().GNOI().Default(t)
+       keyslc := map[string]string{"name": removableLinecard}
        rebootSubComponentRequest := &spb.RebootRequest{
                Method: spb.RebootMethod_COLD,
                Subcomponents: []*tpb.Path{
                        {
-                               Elem: []*tpb.PathElem{{Name: removableLinecard}},
+
+                               Origin: "openconfig-platform",
+                               Elem:   []*tpb.PathElem{{Name: "components"}, {Name: "component", Key: keyslc}, {Name: "state"}, {Name: "location"}},
                        },
                },
        }
