diff --git a/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go b/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go
index f5a2ad14..fed91d01 100644
--- a/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go
+++ b/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go
@@ -78,8 +78,8 @@ func TestMain(m *testing.M) {
 
 func TestStandbyControllerCardReboot(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
-
-	supervisors := components.FindComponentsByType(t, dut, controlcardType)
+	supervisors := []string{"0/RP0/CPU0","0/RP1/CPU0"}
+	//supervisors := components.FindComponentsByType(t, dut, controlcardType)
 	t.Logf("Found supervisor list: %v", supervisors)
 	// Only perform the standby RP rebooting for the chassis with dual RPs/Supervisors.
 	if len(supervisors) != 2 {
@@ -90,12 +90,11 @@ func TestStandbyControllerCardReboot(t *testing.T) {
 	t.Logf("Detected rpStandby: %v, rpActive: %v", rpStandby, rpActive)
 
 	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	compPath := components.GetSubcomponentPath(rpStandby)
 	rebootSubComponentRequest := &spb.RebootRequest{
 		Method: spb.RebootMethod_COLD,
 		Subcomponents: []*tpb.Path{
-			{
-				Elem: []*tpb.PathElem{{Name: rpStandby}},
-			},
+			compPath,
 		},
 	}
 
@@ -123,7 +122,8 @@ func TestLinecardReboot(t *testing.T) {
 	const linecardBoottime = 10 * time.Minute
 	dut := ondatra.DUT(t, "dut")
 
-	lcs := components.FindComponentsByType(t, dut, linecardType)
+	//lcs := components.FindComponentsByType(t, dut, linecardType)
+	lcs := []string{"0/7/CPU0"}
 	t.Logf("Found linecard list: %v", lcs)
 	if got := len(lcs); got == 0 {
 		t.Errorf("Get number of Linecards on %v: got %v, want > 0", dut.Model(), got)
@@ -147,12 +147,11 @@ func TestLinecardReboot(t *testing.T) {
 	}
 
 	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	lineCardPath := components.GetSubcomponentPath(removableLinecard)
 	rebootSubComponentRequest := &spb.RebootRequest{
 		Method: spb.RebootMethod_COLD,
 		Subcomponents: []*tpb.Path{
-			{
-				Elem: []*tpb.PathElem{{Name: removableLinecard}},
-			},
+				lineCardPath,
 		},
 	}
 
@@ -188,6 +187,7 @@ func TestLinecardReboot(t *testing.T) {
 	dut.Telemetry().Component(removableLinecard).Removable().Await(t, linecardBoottime, true)
 
 	t.Logf("Validate interface OperStatus.")
+	time.Sleep(60*time.Second)
 	batch := dut.Telemetry().NewBatch()
 	for _, port := range intfsOperStatusUPBeforeReboot {
 		dut.Telemetry().Interface(port).OperStatus().Batch(t, batch)
