diff --git a/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go b/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go
index 304fe2d7..381803f0 100644
--- a/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go
+++ b/feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go
@@ -80,7 +80,7 @@ func TestStandbyControllerCardReboot(t *testing.T) {
 
 	supervisors := findComponentsByType(t, dut, controlcardType)
 	t.Logf("Found supervisor list: %v", supervisors)
-	// Only perform the standby RP rebooting for the chassis with dual RPs/Supervisors.
+	//Only perform the standby RP rebooting for the chassis with dual RPs/Supervisors.
 	if len(supervisors) != 2 {
 		t.Skipf("Dual RP/SUP is required on %v: got %v, want 2", dut.Model(), len(supervisors))
 	}
@@ -89,11 +89,14 @@ func TestStandbyControllerCardReboot(t *testing.T) {
 	t.Logf("Detected rpStandby: %v, rpActive: %v", rpStandby, rpActive)
 
 	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	keys := map[string]string{"name": rpStandby}
 	rebootSubComponentRequest := &spb.RebootRequest{
 		Method: spb.RebootMethod_COLD,
 		Subcomponents: []*tpb.Path{
+
 			{
-				Elem: []*tpb.PathElem{{Name: rpStandby}},
+				Origin: "openconfig",
+				Elem:   []*tpb.PathElem{{Name: "components"}, {Name: "component", Key: keys}},
 			},
 		},
 	}
@@ -145,12 +148,15 @@ func TestLinecardReboot(t *testing.T) {
 		t.Fatalf("Component(lc).Removable().Get(t): got none, want non-empty")
 	}
 
+	keyslc := map[string]string{"name": removableLinecard}
 	gnoiClient := dut.RawAPIs().GNOI().Default(t)
 	rebootSubComponentRequest := &spb.RebootRequest{
 		Method: spb.RebootMethod_COLD,
 		Subcomponents: []*tpb.Path{
 			{
-				Elem: []*tpb.PathElem{{Name: removableLinecard}},
+
+				Origin: "openconfig",
+				Elem:   []*tpb.PathElem{{Name: "components"}, {Name: "component", Key: keyslc}},
 			},
 		},
 	}
