diff --git a/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go b/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go
index 45b4599b..2c794cd7 100644
--- a/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go
+++ b/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go
@@ -16,6 +16,7 @@ package route_ack_test
 
 import (
 	"context"
+	"flag"
 	"testing"
 	"time"
 
@@ -92,6 +93,10 @@ var (
 	}
 )
 
+var (
+	protocolName  = flag.String("protocol_name", "default", "Name of the protocol to configure static route")
+)
+
 // configInterfaceDUT configures the interface with the Addrs.
 func configInterfaceDUT(i *telemetry.Interface, a *attrs.Attributes) *telemetry.Interface {
 	i.Description = ygot.String(a.Desc)
@@ -203,7 +208,7 @@ func configStaticRoute(t *testing.T, dut *ondatra.DUTDevice, prefix string, next
 	ni1.Enabled = ygot.Bool(true)
 	ni1.Name = ygot.String(*deviations.DefaultNetworkInstance)
 	ni1.Description = ygot.String("Static route added by gNMI-OC")
-	static := ni1.GetOrCreateProtocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, "static-1")
+	static := ni1.GetOrCreateProtocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *protocolName)
 	static.Enabled = ygot.Bool(true)
 	static.Identifier = telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC
 	sr := static.GetOrCreateStatic(prefix)
@@ -250,10 +255,11 @@ func TestRouteAck(t *testing.T) {
 	top.Push(t).StartProtocols(t)
 
 	// Configure the DUT with static route 203.0.113.0/24
-	configureNetworkInstance(t)
+	// Waiting for bug fix: CSCwc27278 [B4 FP: TE-15.1] setting vrf type failes for OC + GNMI
+	//configureNetworkInstance(t)
 	t.Logf("Configure the DUT with static route 203.0.113.0/24...")
 	dutConf := configStaticRoute(t, dut, ateDstNetCIDR, staticNH)
-	dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, "STATIC").Static(ateDstNetCIDR).Replace(t, dutConf)
+	dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *protocolName).Static(ateDstNetCIDR).Replace(t, dutConf)
 	// Verify the entry for 203.0.113.0/24 is active through AFT Telemetry.
 	ipv4Path := dut.Telemetry().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().Ipv4Entry(ateDstNetCIDR)
 	if got, ok := ipv4Path.Prefix().Watch(t, time.Minute, func(val *telemetry.QualifiedString) bool {
