diff --git a/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go b/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go
index 5da175bb..431268eb 100644
--- a/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go
+++ b/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go
@@ -16,6 +16,7 @@ package route_ack_test
 
 import (
 	"context"
+	"flag"
 	"testing"
 	"time"
 
@@ -30,6 +31,10 @@ import (
 	"github.com/openconfig/ygot/ygot"
 )
 
+var (
+	protocolName = flag.String("protocol_name", "default", "Name of the protocol to configure static route")
+)
+
 func TestMain(m *testing.M) {
 	fptest.RunTests(m)
 }
@@ -199,8 +204,9 @@ func configureNetworkInstance(t *testing.T) {
 // configStaticRoute configures a static route.
 func configStaticRoute(t *testing.T, dut *ondatra.DUTDevice, prefix string, nexthop string) *telemetry.NetworkInstance_Protocol {
 	ni1 := dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance).Get(t)
-	static := ni1.GetOrCreateProtocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, "DEFAULT")
-	static.Enabled = ygot.Bool(true)
+	static := ni1.GetOrCreateProtocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *protocolName)
+	// Waiting for bug fix: [B4 FP: TE-15.1, TE-3.7] add support for /network-instances/network-instance/enabled
+	//static.Enabled = ygot.Bool(true)
 	static.Identifier = telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC
 	sr := static.GetOrCreateStatic(prefix)
 	nh := sr.GetOrCreateNextHop("nhg1")
@@ -246,10 +252,12 @@ func TestRouteAck(t *testing.T) {
 	top.Push(t).StartProtocols(t)
 
 	// Configure the DUT with static route 203.0.113.0/24
-	configureNetworkInstance(t)
+	// Waiting for bug fix: CSCwc27278 [B4 FP: TE-15.1] setting vrf type failes for OC + GNMI
+	// Waiting for bug fix: [B4 FP: TE-15.1, TE-3.7] add support for /network-instances/network-instance/enabled
+	//configureNetworkInstance(t)
 	t.Logf("Configure the DUT with static route 203.0.113.0/24...")
 	dutConf := configStaticRoute(t, dut, ateDstNetCIDR, staticNH)
-	dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, "DEFAULT").Update(t, dutConf)
+	dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *protocolName).Update(t, dutConf)
 	// Verify the entry for 203.0.113.0/24 is active through AFT Telemetry.
 	ipv4Path := dut.Telemetry().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().Ipv4Entry(ateDstNetCIDR)
 	if got, ok := ipv4Path.Prefix().Watch(t, time.Minute, func(val *telemetry.QualifiedString) bool {
