diff --git a/feature/gribi/ate_tests/route_removal_non_default_vrf_test/route_removal_non_default_vrf_test.go b/feature/gribi/ate_tests/route_removal_non_default_vrf_test/route_removal_non_default_vrf_test.go
index 0adcbc5f..52573d43 100644
--- a/feature/gribi/ate_tests/route_removal_non_default_vrf_test/route_removal_non_default_vrf_test.go
+++ b/feature/gribi/ate_tests/route_removal_non_default_vrf_test/route_removal_non_default_vrf_test.go
@@ -104,7 +104,8 @@ func TestRouteRemovalNonDefaultVRFFlush(t *testing.T) {
 
 	ateTop.Push(t).StartProtocols(t)
 	configureNetworkInstance(t, dut)
-	configureDUT(t, dut)
+	// will open a bug for this, replacing interface after adding it to vrf causes failure (feature interaction issue).
+	//configureDUT(t, dut)
 
 	// Configure the gRIBI client clientA with election ID of 10.
 	clientA := &gribi.Client{
@@ -303,9 +304,17 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 
 	p1 := dut.Port(t, "port1")
 	p2 := dut.Port(t, "port2")
+	// will communicate this with google, we can not have one deviation to manage several leaves.
+	p1Intf := dutPort1.NewInterface(p1.Name())
+	p1Intf.Enabled = ygot.Bool(true)
+	p1Intf.Subinterface[0].Enabled = ygot.Bool(true)
+
+	p2Intf := dutPort2.NewInterface(p2.Name())
+	p2Intf.Enabled = ygot.Bool(true)
+	p2Intf.Subinterface[0].Enabled = ygot.Bool(true)
 
-	d.Interface(p1.Name()).Replace(t, dutPort1.NewInterface(p1.Name()))
-	d.Interface(p2.Name()).Replace(t, dutPort2.NewInterface(p2.Name()))
+	d.Interface(p1.Name()).Replace(t, p1Intf)
+	d.Interface(p2.Name()).Replace(t, p2Intf)
 
 }
 
