diff --git a/feature/interface/staticarp/ate_tests/static_arp_test/static_arp_test.go b/feature/interface/staticarp/ate_tests/static_arp_test/static_arp_test.go
index a63cac0d..f9def18f 100644
--- a/feature/interface/staticarp/ate_tests/static_arp_test/static_arp_test.go
+++ b/feature/interface/staticarp/ate_tests/static_arp_test/static_arp_test.go
@@ -208,7 +208,8 @@ func testFlow(
 		WithSrcEndpoints(i1).
 		WithDstEndpoints(i2).
 		WithHeaders(headers...).
-		WithEgressTrackingEnabled(33 /* bit offset */, 15 /* width */)
+		// Added due to our Testbed Limitaion
+		WithEgressTrackingEnabled(33 /* bit offset */, 13 /* width */)
 
 	ate.Traffic().Start(t, flow)
 	time.Sleep(15 * time.Second)
