diff --git a/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go b/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
index d68de20c..8f0f77ee 100644
--- a/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
+++ b/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
@@ -88,9 +88,12 @@ func TestSupervisorSwitchover(t *testing.T) {
        }

        gnoiClient := dut.RawAPIs().GNOI().Default(t)
+       keys := map[string]string{"name": rpStandbyBeforeSwitch}
        switchoverRequest := &spb.SwitchControlProcessorRequest{
                ControlProcessor: &tpb.Path{
-                       Elem: []*tpb.PathElem{{Name: rpStandbyBeforeSwitch}},
+
+                       Origin: "openconfig",
+                       Elem:   []*tpb.PathElem{{Name: "components"}, {Name: "component", Key: keys}, {Name: "state"}, {Name: "location"}},
                },
        }
        t.Logf("switchoverRequest: %v", switchoverRequest)
