diff --git a/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go b/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
index e9b3ce1..dd04245 100644
--- a/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
+++ b/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
@@ -16,16 +16,17 @@ package supervisor_switchover_test
 import (
 	"context"
 	"sort"
+	"strings"
 	"testing"
 	"time"
 
 	"github.com/openconfig/featureprofiles/internal/components"
+	"github.com/openconfig/featureprofiles/internal/deviations"
 	"github.com/openconfig/featureprofiles/internal/fptest"
 	"github.com/openconfig/ondatra"
 	"github.com/openconfig/testt"
 
 	spb "github.com/openconfig/gnoi/system"
-	tpb "github.com/openconfig/gnoi/types"
 	"github.com/openconfig/ondatra/gnmi"
 	"github.com/openconfig/ondatra/gnmi/oc"
 	"github.com/openconfig/ygnmi/ygnmi"
@@ -78,7 +79,6 @@ func TestSupervisorSwitchover(t *testing.T) {
 	if got, want := len(supervisors), 2; got != want {
 		t.Skipf("Dual RP/SUP is required on %v: got %v, want %v", dut.Model(), got, want)
 	}
-
 	rpStandbyBeforeSwitch, rpActiveBeforeSwitch := findStandbyRP(t, dut, supervisors)
 	t.Logf("Detected rpStandby: %v, rpActive: %v", rpStandbyBeforeSwitch, rpActiveBeforeSwitch)
 
@@ -97,9 +97,7 @@ func TestSupervisorSwitchover(t *testing.T) {
 
 	gnoiClient := dut.RawAPIs().GNOI().New(t)
 	switchoverRequest := &spb.SwitchControlProcessorRequest{
-		ControlProcessor: &tpb.Path{
-			Elem: []*tpb.PathElem{{Name: rpStandbyBeforeSwitch}},
-		},
+		ControlProcessor: components.GetSubcomponentPath(rpStandbyBeforeSwitch),
 	}
 	t.Logf("switchoverRequest: %v", switchoverRequest)
 	switchoverResponse, err := gnoiClient.System().SwitchControlProcessor(context.Background(), switchoverRequest)
@@ -109,7 +107,13 @@ func TestSupervisorSwitchover(t *testing.T) {
 	t.Logf("gnoiClient.System().SwitchControlProcessor() response: %v, err: %v", switchoverResponse, err)
 
 	want := rpStandbyBeforeSwitch
-	if got := switchoverResponse.GetControlProcessor().GetElem()[0].GetName(); got != want {
+	got := ""
+	if *deviations.GNOISubcomponentPath {
+		got = switchoverResponse.GetControlProcessor().GetElem()[0].GetName()
+	} else {
+		got = switchoverResponse.GetControlProcessor().GetElem()[1].GetKey()["name"]
+	}
+	if got != want {
 		t.Fatalf("switchoverResponse.GetControlProcessor().GetElem()[0].GetName(): got %v, want %v", got, want)
 	}
 	if got, want := switchoverResponse.GetVersion(), ""; got == want {
@@ -149,6 +153,13 @@ func TestSupervisorSwitchover(t *testing.T) {
 		t.Errorf("Get rpStandbyAfterSwitch: got %v, want %v", got, want)
 	}
 
+	interfaceCheckToSkip := ""
+	if *deviations.SkipStandbyManagementInterfaceCheck {
+		if dut.Vendor() == ondatra.CISCO {
+			interfaceCheckToSkip = "MgmtEth" + rpStandbyAfterSwitch
+		}
+	}
+
 	batch := gnmi.OCBatch()
 	for _, port := range intfsOperStatusUPBeforeSwitch {
 		batch.AddPaths(gnmi.OC().Interface(port).OperStatus())
@@ -159,6 +170,9 @@ func TestSupervisorSwitchover(t *testing.T) {
 			return false
 		}
 		for _, port := range intfsOperStatusUPBeforeSwitch {
+			if strings.HasPrefix(port, interfaceCheckToSkip) {
+				continue
+			}
 			if root.GetInterface(port).GetOperStatus() != oc.Interface_OperStatus_UP {
 				return false
 			}
diff --git a/internal/deviations/deviations.go b/internal/deviations/deviations.go
index 6e990bb..4c0cac2 100644
--- a/internal/deviations/deviations.go
+++ b/internal/deviations/deviations.go
@@ -111,4 +111,7 @@ var (
 
 	ExplicitInterfaceInDefaultVRF = flag.Bool("deviation_explicit_interface_in_default_vrf", false,
 		"Device requires explicit attachment of an interface or subinterface to the default network instance. OpenConfig expects an unattached interface or subinterface to be implicitly part of the default network instance. Fully-compliant devices should pass with and without this deviation.")
+
+	SkipStandbyManagementInterfaceCheck = flag.Bool("deviation_skip_standby_management_interface_check", true,
+		"Device does not generate an event for backup menegment interface after switchover, so suppress the oper-status check for backup managment interface after switchover.")
 )
