diff --git a/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go b/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
index c79b11dc..2f1d3dcd 100644
--- a/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
+++ b/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
@@ -87,9 +87,11 @@ func TestSupervisorSwitchover(t *testing.T) {
 	}
 
 	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	keys := map[string]string{"name": rpStandbyBeforeSwitch}
 	switchoverRequest := &spb.SwitchControlProcessorRequest{
 		ControlProcessor: &tpb.Path{
-			Elem: []*tpb.PathElem{{Name: rpStandbyBeforeSwitch}},
+			Origin: "openconfig",
+			Elem:   []*tpb.PathElem{{Name: "components"}, {Name: "component", Key: keys}},
 		},
 	}
 	t.Logf("switchoverRequest: %v", switchoverRequest)
