diff --git a/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go b/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
index e9b3ce16..91a6da20 100644
--- a/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
+++ b/feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go
@@ -20,12 +20,12 @@ import (
 	"time"
 
 	"github.com/openconfig/featureprofiles/internal/components"
+	"github.com/openconfig/featureprofiles/internal/deviations"
 	"github.com/openconfig/featureprofiles/internal/fptest"
 	"github.com/openconfig/ondatra"
 	"github.com/openconfig/testt"
 
 	spb "github.com/openconfig/gnoi/system"
-	tpb "github.com/openconfig/gnoi/types"
 	"github.com/openconfig/ondatra/gnmi"
 	"github.com/openconfig/ondatra/gnmi/oc"
 	"github.com/openconfig/ygnmi/ygnmi"
@@ -73,12 +73,12 @@ func TestSupervisorSwitchover(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
 
 	supervisors := components.FindComponentsByType(t, dut, controlcardType)
+	//supervisors := []string{"0/RP0/CPU0","0/RP1/CPU0"}
 	t.Logf("Found supervisor list: %v", supervisors)
 	// Only perform the switchover for the chassis with dual RPs/Supervisors.
 	if got, want := len(supervisors), 2; got != want {
 		t.Skipf("Dual RP/SUP is required on %v: got %v, want %v", dut.Model(), got, want)
 	}
-
 	rpStandbyBeforeSwitch, rpActiveBeforeSwitch := findStandbyRP(t, dut, supervisors)
 	t.Logf("Detected rpStandby: %v, rpActive: %v", rpStandbyBeforeSwitch, rpActiveBeforeSwitch)
 
@@ -96,10 +96,9 @@ func TestSupervisorSwitchover(t *testing.T) {
 	}
 
 	gnoiClient := dut.RawAPIs().GNOI().New(t)
+	compPath := components.GetSubcomponentPath(rpStandbyBeforeSwitch)
 	switchoverRequest := &spb.SwitchControlProcessorRequest{
-		ControlProcessor: &tpb.Path{
-			Elem: []*tpb.PathElem{{Name: rpStandbyBeforeSwitch}},
-		},
+		ControlProcessor: compPath, 
 	}
 	t.Logf("switchoverRequest: %v", switchoverRequest)
 	switchoverResponse, err := gnoiClient.System().SwitchControlProcessor(context.Background(), switchoverRequest)
@@ -109,7 +108,13 @@ func TestSupervisorSwitchover(t *testing.T) {
 	t.Logf("gnoiClient.System().SwitchControlProcessor() response: %v, err: %v", switchoverResponse, err)
 
 	want := rpStandbyBeforeSwitch
-	if got := switchoverResponse.GetControlProcessor().GetElem()[0].GetName(); got != want {
+	got:= ""
+	if *deviations.GNOISubcomponentPath {
+		got = switchoverResponse.GetControlProcessor().GetElem()[0].GetName()
+	} else {
+		got = switchoverResponse.GetControlProcessor().GetElem()[1].GetKey()["name"]
+	}
+	if got != want {
 		t.Fatalf("switchoverResponse.GetControlProcessor().GetElem()[0].GetName(): got %v, want %v", got, want)
 	}
 	if got, want := switchoverResponse.GetVersion(), ""; got == want {
@@ -148,6 +153,13 @@ func TestSupervisorSwitchover(t *testing.T) {
 	if got, want := rpStandbyAfterSwitch, rpActiveBeforeSwitch; got != want {
 		t.Errorf("Get rpStandbyAfterSwitch: got %v, want %v", got, want)
 	}
+    
+	switchoverReady = gnmi.OC().Component(rpActiveBeforeSwitch).SwitchoverReady()
+	gnmi.Await(t, dut, switchoverReady.State(), 30*time.Minute, true)
+	t.Logf("SwitchoverReady().Get(t): %v", gnmi.Get(t, dut, switchoverReady.State()))
+	if got, want := gnmi.Get(t, dut, switchoverReady.State()), true; got != want {
+		t.Errorf("switchoverReady.Get(t): got %v, want %v", got, want)
+	}
 
 	batch := gnmi.OCBatch()
 	for _, port := range intfsOperStatusUPBeforeSwitch {
