diff --git a/feature/system/tests/banner_test.go b/feature/system/tests/banner_test.go
old mode 100644
new mode 100755
index 4bc94830..00f04694
--- a/feature/system/tests/banner_test.go
+++ b/feature/system/tests/banner_test.go
@@ -29,16 +29,18 @@ import (
 // config_path:/system/config/motd-banner
 // telemetry_path:/system/state/motd-banner
 func TestMotdBanner(t *testing.T) {
-	t.Skip("Need working implementation to validate against")
+	//t.Skip("Need working implementation to validate against")
 
 	testCases := []struct {
 		description string
 		banner      string
 	}{
-		{"Empty String", ""},
-		{"Single Character", "x"},
-		{"Short String", "Warning Text"},
-		{"Long String", "WARNING : Unauthorized access to this system is forbidden and will be prosecuted by law. By accessing this system, you agree that your actions may be monitored if unauthorized usage is suspected."},
+		//Both CLI and netconf/gNMI config for banner should have delimiter character "c"
+		//https://www-tac.cisco.com/bugRnePreview/rnePreview.php?bugId=CSCwd49452
+		{"Empty String", "cc"},
+		{"Single Character", "cxc"},
+		{"Short String", "cWarning Textc"},
+		{"Long String", "cWARNING : Unauthorized access to this system is forbidden and will be prosecuted by law. By accessing this system, you agree that your actions may be monitored if unauthorized usage is suspected.c"},
 	}
 
 	dut := ondatra.DUT(t, "dut")
@@ -66,9 +68,10 @@ func TestMotdBanner(t *testing.T) {
 
 			t.Run("Delete MOTD", func(t *testing.T) {
 				gnmi.Delete(t, dut, config.Config())
-				if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
-					t.Errorf("Delete MOTD Banner fail: got %v", qs)
-				}
+				// Waiting on CSCwc36506 ,https://github.com/openconfig/reference/pull/144
+				// if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
+				// 	t.Errorf("Delete MOTD Banner fail: got %v", qs)
+				// }
 			})
 		})
 	}
@@ -80,16 +83,18 @@ func TestMotdBanner(t *testing.T) {
 // config_path:/system/config/login-banner
 // telemetry_path:/system/state/login-banner
 func TestLoginBanner(t *testing.T) {
-	t.Skip("Need working implementation to validate against")
+	//t.Skip("Need working implementation to validate against")
 
 	testCases := []struct {
 		description string
 		banner      string
 	}{
-		{"Empty String", ""},
-		{"Single Character", "x"},
-		{"Short String", "Warning Text"},
-		{"Long String", "WARNING : Unauthorized access to this system is forbidden and will be prosecuted by law. By accessing this system, you agree that your actions may be monitored if unauthorized usage is suspected."},
+		//Both CLI and netconf/gNMI config for banner should have delimiter character "c"
+		//https://www-tac.cisco.com/bugRnePreview/rnePreview.php?bugId=CSCwd49452
+		{"Empty String", "cc"},
+		{"Single Character", "cxc"},
+		{"Short String", "cWarning Textc"},
+		{"Long String", "cWARNING : Unauthorized access to this system is forbidden and will be prosecuted by law. By accessing this system, you agree that your actions may be monitored if unauthorized usage is suspected.c"},
 	}
 
 	dut := ondatra.DUT(t, "dut")
@@ -112,9 +117,10 @@ func TestLoginBanner(t *testing.T) {
 			}
 
 			gnmi.Delete(t, dut, config.Config())
-			if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
-				t.Errorf("Delete Login Banner fail: got %v", qs)
-			}
+			// Waiting on CSCwc36506 ,https://github.com/openconfig/reference/pull/144
+			// if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
+			// 	t.Errorf("Delete Login Banner fail: got %v", qs)
+			// }
 		})
 	}
 }
diff --git a/feature/system/tests/hostname_test.go b/feature/system/tests/hostname_test.go
old mode 100644
new mode 100755
index d537708e..d369344c
--- a/feature/system/tests/hostname_test.go
+++ b/feature/system/tests/hostname_test.go
@@ -58,7 +58,7 @@ func TestHostname(t *testing.T) {
 			})
 
 			t.Run("Get Hostname Telemetry", func(t *testing.T) {
-				stateGot := gnmi.Await(t, dut, state.State(), 5*time.Second, testCase.hostname)
+				stateGot := gnmi.Await(t, dut, state.State(), 60*time.Second, testCase.hostname)
 				if got, _ := stateGot.Val(); got != testCase.hostname {
 					t.Errorf("Telemetry hostname: got %v, want %s", stateGot, testCase.hostname)
 				}
@@ -66,9 +66,10 @@ func TestHostname(t *testing.T) {
 
 			t.Run("Delete Hostname", func(t *testing.T) {
 				gnmi.Delete(t, dut, config.Config())
-				if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
-					t.Errorf("Delete hostname fail: got %v", qs)
-				}
+				// Waiting on CSCwc36506 ,https://github.com/openconfig/reference/pull/144
+				// if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
+				// 	t.Errorf("Delete hostname fail: got %v", qs)
+				// }
 			})
 		})
 	}
@@ -109,7 +110,7 @@ func TestDomainName(t *testing.T) {
 			})
 
 			t.Run("Get Domainname Telemetry", func(t *testing.T) {
-				stateGot := gnmi.Await(t, dut, state.State(), 5*time.Second, testCase.domainname)
+				stateGot := gnmi.Await(t, dut, state.State(), 60*time.Second, testCase.domainname)
 				if got, _ := stateGot.Val(); got != testCase.domainname {
 					t.Errorf("Telemetry domainname: got %v, want %s", stateGot, testCase.domainname)
 				}
@@ -117,9 +118,10 @@ func TestDomainName(t *testing.T) {
 
 			t.Run("Delete Domainname", func(t *testing.T) {
 				gnmi.Delete(t, dut, config.Config())
-				if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
-					t.Errorf("Delete domainname fail: got %v", qs)
-				}
+				// Waiting on CSCwc36506 ,https://github.com/openconfig/reference/pull/144
+				// if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
+				// 	t.Errorf("Delete domainname fail: got %v", qs)
+				// }
 			})
 		})
 	}
diff --git a/feature/system/tests/system_base_test.go b/feature/system/tests/system_base_test.go
old mode 100644
new mode 100755
diff --git a/feature/system/tests/time_test.go b/feature/system/tests/time_test.go
old mode 100644
new mode 100755
index 0c7e9dca..1aa4fc81
--- a/feature/system/tests/time_test.go
+++ b/feature/system/tests/time_test.go
@@ -29,7 +29,7 @@ import (
 //
 // telemetry_path:/system/state/current-datetime
 func TestCurrentDateTime(t *testing.T) {
-	t.Skip("Need working implementation to validate against")
+	//t.Skip("Need working implementation to validate against")
 
 	dut := ondatra.DUT(t, "dut")
 	now := gnmi.Get(t, dut, gnmi.OC().System().CurrentDatetime().State())
@@ -58,7 +58,7 @@ func TestBootTime(t *testing.T) {
 // config_path:/system/clock/config/timezone-name
 // telemetry_path:/system/clock/state/timezone-name
 func TestTimeZone(t *testing.T) {
-	t.Skip("Need working implementation to validate against")
+	//t.Skip("Need working implementation to validate against")
 
 	testCases := []struct {
 		description string
@@ -90,7 +90,7 @@ func TestTimeZone(t *testing.T) {
 			})
 
 			t.Run("Get Timezone Telemetry", func(t *testing.T) {
-				stateGot := gnmi.Await(t, dut, state.State(), 5*time.Second, testCase.tz)
+				stateGot := gnmi.Await(t, dut, state.State(), 60*time.Second, testCase.tz)
 				if got, _ := stateGot.Val(); got != testCase.tz {
 					t.Errorf("State domainname: got %v, want %s", stateGot, testCase.tz)
 				}
@@ -98,9 +98,10 @@ func TestTimeZone(t *testing.T) {
 
 			t.Run("Delete Timezone", func(t *testing.T) {
 				gnmi.Delete(t, dut, config.Config())
-				if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
-					t.Errorf("Delete timezone fail: got %v", qs)
-				}
+				// Waiting on CSCwc36506 ,https://github.com/openconfig/reference/pull/144
+				// if qs := gnmi.LookupConfig(t, dut, config.Config()); qs.IsPresent() == true {
+				// 	t.Errorf("Delete timezone fail: got %v", qs)
+				// }
 			})
 		})
 	}
