diff --git a/feature/gnmi/ate_tests/telemetry_port_speed_test/telemetry_port_speed_test.go b/feature/gnmi/ate_tests/telemetry_port_speed_test/telemetry_port_speed_test.go
index c7dca01a..dbc8ceb0
--- a/feature/gnmi/ate_tests/telemetry_port_speed_test/telemetry_port_speed_test.go
+++ b/feature/gnmi/ate_tests/telemetry_port_speed_test/telemetry_port_speed_test.go
@@ -191,7 +191,7 @@ func (tc *testCase) configureDUT(t *testing.T) {
 		fptest.LogYgot(t, "LACP", lacpPath, lacp)
 		lacpPath.Replace(t, lacp)
 		t.Cleanup(func() {
-			lacpPath.Delete(t)
+			d.Lacp().Delete(t)
 		})
 	}
 
