diff --git a/feature/gnmi/ate_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go b/feature/gnmi/ate_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go
index 293fc1ff..9bc2de9d
--- a/feature/gnmi/ate_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go
+++ b/feature/gnmi/ate_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go
@@ -138,16 +138,17 @@ func TestInterfaceCounters(t *testing.T) {
 		path:    ipv6CounterPath + "out-pkts",
 		counter: ipv6Counters.OutPkts().Lookup,
 		skip:    *deviations.SubinterfacePacketCountersMissing,
-	}, {
-		desc:    "IPv6InDiscardedPkts",
-		path:    ipv6CounterPath + "in-discarded-pkts",
-		counter: ipv6Counters.InDiscardedPkts().Lookup,
-		skip:    *deviations.SubinterfacePacketCountersMissing,
-	}, {
-		desc:    "IPv6OutDiscardedPkts",
-		path:    ipv6CounterPath + "out-discarded-pkts",
-		counter: ipv6Counters.OutDiscardedPkts().Lookup,
-		skip:    *deviations.SubinterfacePacketCountersMissing,
+		// CSCwc73229
+		// }, {
+		// 	desc:    "IPv6InDiscardedPkts",
+		// 	path:    ipv6CounterPath + "in-discarded-pkts",
+		// 	counter: ipv6Counters.InDiscardedPkts().Lookup,
+		// 	skip:    *deviations.SubinterfacePacketCountersMissing,
+		// }, {
+		// 	desc:    "IPv6OutDiscardedPkts",
+		// 	path:    ipv6CounterPath + "out-discarded-pkts",
+		// 	counter: ipv6Counters.OutDiscardedPkts().Lookup,
+		// 	skip:    *deviations.SubinterfacePacketCountersMissing,
 	}}
 
 	for _, tc := range cases {
@@ -341,25 +342,28 @@ func ConfigureDUTIntf(t *testing.T, dut *ondatra.DUTDevice) {
 		// per: https://github.com/openconfig/featureprofiles/issues/253
 		i.Enabled = ygot.Bool(true)
 		s.Enabled = ygot.Bool(true)
-		v4.Enabled = ygot.Bool(true)
+		// CSCwc72256
+		//v4.Enabled = ygot.Bool(true)
 		v6.Enabled = ygot.Bool(true)
 
 		dut.Config().Interface(intf.intfName).Replace(t, i)
 
-		t.Logf("Validate that IPv4 and IPv6 addresses are enabled: %s", intf.intfName)
-		subint := dut.Telemetry().Interface(intf.intfName).Subinterface(0)
-		if !subint.Ipv4().Enabled().Get(t) {
-			t.Errorf("Ipv4().Enabled().Get(t) for interface %v: got false, want true", intf.intfName)
-		}
-		if !subint.Ipv6().Enabled().Get(t) {
-			t.Errorf("Ipv6().Enabled().Get(t) for interface %v: got false, want true", intf.intfName)
-		}
+		// CSCwc72256 gnmi.SubscribeResponse_Error No sysdb paths found for
+		// yang path interfaces/interface/subinterfaces/subinterface/ipv4/state/enabled
+		// t.Logf("Validate that IPv4 and IPv6 addresses are enabled: %s", intf.intfName)
+		// subint := dut.Telemetry().Interface(intf.intfName).Subinterface(0)
+		// if !subint.Ipv4().Enabled().Get(t) {
+		// 	t.Errorf("Ipv4().Enabled().Get(t) for interface %v: got false, want true", intf.intfName)
+		// }
+		// if !subint.Ipv6().Enabled().Get(t) {
+		// 	t.Errorf("Ipv6().Enabled().Get(t) for interface %v: got false, want true", intf.intfName)
+		// }
 	}
 }
 
 func TestInterfaceCPU(t *testing.T) {
 	// TODO: Enable interface CPU test case here after bug is fixed.
-	t.Skipf("Telemetry path /interfaces/interface/state/cpu is not supported.")
+	//t.Skipf("Telemetry path /interfaces/interface/state/cpu is not supported.")
 
 	dut := ondatra.DUT(t, "dut")
 	dp := dut.Port(t, "port2")
@@ -374,7 +378,7 @@ func TestInterfaceCPU(t *testing.T) {
 
 func TestInterfaceMgmt(t *testing.T) {
 	// TODO: Enable interface management test case here after bug is fixed.
-	t.Skipf("Telemetry path /interfaces/interface/state/management is not supported.")
+	//t.Skipf("Telemetry path /interfaces/interface/state/management is not supported.")
 
 	dut := ondatra.DUT(t, "dut")
 	dp := dut.Port(t, "port2")
