diff --git a/feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go b/feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go
index 581e0771..9ac86620
--- a/feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go
+++ b/feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go
@@ -15,32 +15,29 @@
 package tls_authentication_over_grpc_test
 
 import (
+	"bytes"
 	"context"
 	"crypto/tls"
+	"encoding/json"
 	"flag"
 	"fmt"
+	"os"
 	"testing"
 
 	"github.com/openconfig/featureprofiles/internal/fptest"
 	"github.com/openconfig/ondatra"
-	"github.com/openconfig/ygot/ygot"
 	"golang.org/x/crypto/ssh"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/metadata"
 
 	gpb "github.com/openconfig/gnmi/proto/gnmi"
-	"github.com/openconfig/ondatra/gnmi"
-	"github.com/openconfig/ondatra/gnmi/oc"
 )
 
 var (
-	sshIP = flag.String("ssh_ip", "", "External IP address of management interface.")
-)
-
-const (
-	sshPort  = 22
-	gnmiPort = 6030
+	sshIP    = flag.String("ssh_ip", "1.1.1.1", "External IP address of management interface.")
+	sshPort  = flag.String("ssh_port", "22", "SSH port")
+	gnmiPort = flag.String("gnmi_nondefault_port", "57777", "Gnmi port")
 )
 
 func TestMain(m *testing.M) {
@@ -59,7 +56,7 @@ func keyboardInteraction(password string) ssh.KeyboardInteractiveChallenge {
 func gnmiClient(ctx context.Context) (gpb.GNMIClient, error) {
 	conn, err := grpc.DialContext(
 		ctx,
-		fmt.Sprintf("%s:%d", *sshIP, gnmiPort),
+		fmt.Sprintf("%s:%s", *sshIP, *gnmiPort),
 		grpc.WithTransportCredentials(
 			credentials.NewTLS(&tls.Config{
 				InsecureSkipVerify: true, // NOLINT
@@ -75,14 +72,59 @@ func TestAuthentication(t *testing.T) {
 	if *sshIP == "" {
 		t.Fatal("--ssh_ip flag must be set.")
 	}
-
 	dut := ondatra.DUT(t, "dut")
-	gnmi.Replace(t, dut, gnmi.OC().System().Aaa().Authentication().
-		User("alice").Config(), &oc.System_Aaa_Authentication_User{
-		Username: ygot.String("alice"),
-		Password: ygot.String("password"),
-		Role:     oc.AaaTypes_SYSTEM_DEFINED_ROLES_SYSTEM_ROLE_ADMIN,
-	})
+	// https://github.com/openconfig/ondatra/issues/50
+	rawData := fmt.Sprintf(`{
+		"openconfig-system:system": {
+		 "aaa": {
+		  "authentication": {
+		   "users": {
+			"user": [
+			 {
+			  "username": "alice",
+			  "config": {
+			   "username": "alice",
+			   "role": "root-lr",
+			   "password": "password"
+			  }
+			 }
+			]
+		   }
+		  }
+		 }
+		}
+	  }`)
+	f, err := os.Create("user.json")
+	if err != nil {
+		t.Error("Error occcured while creating the file ", err)
+	}
+	defer f.Close()
+	defer os.Remove("user.json")
+	_, err2 := f.WriteString(rawData)
+	if err2 != nil {
+		t.Error("Error occcured while writing to file ", err2)
+	}
+	client := dut.RawAPIs().GNMI().New(t)
+	data, _ := os.ReadFile("user.json")
+	rawjson := string(bytes.TrimRight(data, "\n"))
+	setreq := &gpb.SetRequest{
+		Update: []*gpb.Update{{
+			Path: &gpb.Path{
+				Origin: "openconfig",
+				Elem:   []*gpb.PathElem{},
+			},
+			Val: &gpb.TypedValue{
+				Value: &gpb.TypedValue_JsonIetfVal{
+					JsonIetfVal: []byte(rawjson),
+				},
+			},
+		}},
+	}
+	response, err := client.Set(context.Background(), setreq)
+	if err != nil {
+		t.Log(response)
+		t.Error(err)
+	}
 
 	tests := []struct {
 		desc       string
@@ -106,7 +148,7 @@ func TestAuthentication(t *testing.T) {
 	for _, tc := range tests {
 		t.Run(tc.desc, func(t *testing.T) {
 			t.Log("Trying SSH credentials")
-			sshClient, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", *sshIP, sshPort), &ssh.ClientConfig{
+			sshClient, err := ssh.Dial("tcp", fmt.Sprintf("%s:%s", *sshIP, *sshPort), &ssh.ClientConfig{
 				User: tc.user,
 				Auth: []ssh.AuthMethod{
 					ssh.KeyboardInteractive(keyboardInteraction(tc.pass)),
@@ -139,7 +181,8 @@ func TestAuthentication(t *testing.T) {
 					Elem: []*gpb.PathElem{
 						{Name: "system"}, {Name: "config"}, {Name: "hostname"}}},
 				},
-				Type: gpb.GetRequest_CONFIG,
+				Type:     gpb.GetRequest_CONFIG,
+				Encoding: gpb.Encoding_JSON_IETF,
 			})
 			if tc.wantErr != (err != nil) {
 				if tc.wantErr {
@@ -149,15 +192,15 @@ func TestAuthentication(t *testing.T) {
 				}
 			}
 			t.Log("Trying credentials with GNMI Set")
+			// Limitation due to CSCwd49452
+			json_config, _ := json.Marshal("cmessage of the dayc")
 			_, err = gnmi.Set(ctx, &gpb.SetRequest{
 				Replace: []*gpb.Update{{
 					Path: &gpb.Path{
 						Elem: []*gpb.PathElem{
 							{Name: "system"}, {Name: "config"}, {Name: "motd-banner"}},
 					},
-					Val: &gpb.TypedValue{
-						Value: &gpb.TypedValue_StringVal{StringVal: "message of the day"},
-					},
+					Val: &gpb.TypedValue{Value: &gpb.TypedValue_JsonIetfVal{JsonIetfVal: json_config}},
 				}},
 			})
 			if tc.wantErr != (err != nil) {
