diff --git a/feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go b/feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go
index afb62b81..260a4dea 100644
--- a/feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go
+++ b/feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go
@@ -20,6 +20,8 @@ import (
 	"encoding/json"
 	"flag"
 	"fmt"
+	"os"
+	"strings"
 	"testing"
 
 	"github.com/openconfig/featureprofiles/internal/deviations"
@@ -32,17 +34,16 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/encoding/prototext"
 
+	bindpb "github.com/openconfig/featureprofiles/topologies/proto/binding"
 	gpb "github.com/openconfig/gnmi/proto/gnmi"
 )
 
 var (
-	sshIP = flag.String("ssh_ip", "", "External IP address of management interface.")
-)
-
-const (
-	sshPort  = 22
-	gnmiPort = 6030
+	sshIP    = flag.String("ssh_ip", "1.1.1.1", "External IP address of management interface.")
+	sshPort  = flag.String("ssh_port", "22", "SSH port")
+	gnmiPort = flag.String("gnmi_nondefault_port", "57777", "Gnmi port")
 )
 
 func TestMain(m *testing.M) {
@@ -61,7 +62,7 @@ func keyboardInteraction(password string) ssh.KeyboardInteractiveChallenge {
 func gnmiClient(ctx context.Context) (gpb.GNMIClient, error) {
 	conn, err := grpc.DialContext(
 		ctx,
-		fmt.Sprintf("%s:%d", *sshIP, gnmiPort),
+		fmt.Sprintf("%s:%s", *sshIP, *gnmiPort),
 		grpc.WithTransportCredentials(
 			credentials.NewTLS(&tls.Config{
 				InsecureSkipVerify: true, // NOLINT
@@ -73,7 +74,37 @@ func gnmiClient(ctx context.Context) (gpb.GNMIClient, error) {
 	return gpb.NewGNMIClient(conn), nil
 }
 
+func parseBindingFile(t *testing.T) (string, string, string) {
+	t.Helper()
+
+	bindingFile := flag.Lookup("binding").Value.String()
+	in, err := os.ReadFile(bindingFile)
+	if err != nil {
+		t.Fatalf("unable to read binding file")
+	}
+
+	b := &bindpb.Binding{}
+	if err := prototext.Unmarshal(in, b); err != nil {
+		t.Fatalf("unable to parse binding file")
+	}
+
+	sshTarget := strings.Split(b.Duts[0].Ssh.Target, ":")
+	gnmiTarget := strings.Split(b.Duts[0].Gnmi.Target, ":")
+
+	sshIp := sshTarget[0]
+	sshPort := "22"
+	gnmiPort := "6030"
+	if len(sshTarget) > 1 {
+		sshPort = sshTarget[1]
+	}
+	if len(gnmiTarget) > 1 {
+		gnmiPort = gnmiTarget[1]
+	}
+	return sshIp, sshPort, gnmiPort
+}
+
 func TestAuthentication(t *testing.T) {
+	*sshIP, *sshPort, *gnmiPort = parseBindingFile(t)
 	if *sshIP == "" {
 		t.Fatal("--ssh_ip flag must be set.")
 	}
@@ -108,7 +139,7 @@ func TestAuthentication(t *testing.T) {
 	for _, tc := range tests {
 		t.Run(tc.desc, func(t *testing.T) {
 			t.Log("Trying SSH credentials")
-			sshClient, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", *sshIP, sshPort), &ssh.ClientConfig{
+			sshClient, err := ssh.Dial("tcp", fmt.Sprintf("%s:%s", *sshIP, *sshPort), &ssh.ClientConfig{
 				User: tc.user,
 				Auth: []ssh.AuthMethod{
 					ssh.KeyboardInteractive(keyboardInteraction(tc.pass)),
