diff --git a/feature/gnoi/system/tests/traceroute_test/traceroute_test.go b/feature/gnoi/system/tests/traceroute_test/traceroute_test.go
index fb5a8b75..b8600d66 100644
--- a/feature/gnoi/system/tests/traceroute_test/traceroute_test.go
+++ b/feature/gnoi/system/tests/traceroute_test/traceroute_test.go
@@ -18,6 +18,7 @@ import (
 	"context"
 	"io"
 	"testing"
+	"time"
 
 	"github.com/openconfig/featureprofiles/internal/deviations"
 	"github.com/openconfig/featureprofiles/internal/fptest"
@@ -30,7 +31,7 @@ import (
 
 const (
 	minTracerouteHops        = 1
-	minTracerouteRTT         = 1
+	minTracerouteRTT         = 0 // the device traceroute to its loopback, the RTT can be zero.
 	maxDefaultTracerouteHops = 30
 )
 
@@ -99,30 +100,35 @@ func TestGNOITraceroute(t *testing.T) {
 		fptest.AssignToNetworkInstance(t, dut, lbIntf, *deviations.DefaultNetworkInstance, 0)
 	}
 	cases := []struct {
-		desc         string
-		traceRequest *spb.TracerouteRequest
+		desc              string
+		traceRequest      *spb.TracerouteRequest
+		defaultL4Protocol bool
 	}{
 		{
-			desc: "Check traceroute with IPv4 destination",
+			desc:              "Check traceroute with IPv4 destination",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv4Addrs[0].GetIp(),
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv6 destination",
+			desc:              "Check traceroute with IPv6 destination",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv6Addrs[0].GetIp(),
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv6 protocol",
+			desc:              "Check traceroute with IPv6 protocol",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv6Addrs[0].GetIp(),
 				L3Protocol:     tpb.L3Protocol_IPV6,
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv4 DF bit",
+			desc:              "Check traceroute with IPv4 DF bit",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv4Addrs[0].GetIp(),
 				L3Protocol:     tpb.L3Protocol_IPV4,
@@ -130,7 +136,8 @@ func TestGNOITraceroute(t *testing.T) {
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv4 do_not_resolve",
+			desc:              "Check traceroute with IPv4 do_not_resolve",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv4Addrs[0].GetIp(),
 				L3Protocol:     tpb.L3Protocol_IPV4,
@@ -138,7 +145,8 @@ func TestGNOITraceroute(t *testing.T) {
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv6 do_not_resolve",
+			desc:              "Check traceroute with IPv6 do_not_resolve",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv6Addrs[0].GetIp(),
 				L3Protocol:     tpb.L3Protocol_IPV6,
@@ -146,7 +154,8 @@ func TestGNOITraceroute(t *testing.T) {
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv4 wait",
+			desc:              "Check traceroute with IPv4 wait",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv4Addrs[0].GetIp(),
 				L3Protocol:     tpb.L3Protocol_IPV4,
@@ -154,7 +163,8 @@ func TestGNOITraceroute(t *testing.T) {
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv6 wait",
+			desc:              "Check traceroute with IPv6 wait",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv6Addrs[0].GetIp(),
 				L3Protocol:     tpb.L3Protocol_IPV6,
@@ -162,7 +172,8 @@ func TestGNOITraceroute(t *testing.T) {
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv4 TTL",
+			desc:              "Check traceroute with IPv4 TTL",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv4Addrs[0].GetIp(),
 				L3Protocol:     tpb.L3Protocol_IPV4,
@@ -171,7 +182,8 @@ func TestGNOITraceroute(t *testing.T) {
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv6 TTL",
+			desc:              "Check traceroute with IPv6 TTL",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv6Addrs[0].GetIp(),
 				L3Protocol:     tpb.L3Protocol_IPV6,
@@ -179,7 +191,8 @@ func TestGNOITraceroute(t *testing.T) {
 				DoNotLookupAsn: true,
 			}},
 		{
-			desc: "Check traceroute with IPv4 L4protocol ICMP",
+			desc:              "Check traceroute with IPv4 L4protocol ICMP",
+			defaultL4Protocol: true,
 			traceRequest: &spb.TracerouteRequest{
 				Destination:    ipv4Addrs[0].GetIp(),
 				L3Protocol:     tpb.L3Protocol_IPV4,
@@ -230,6 +243,20 @@ func TestGNOITraceroute(t *testing.T) {
 	gnoiClient := dut.RawAPIs().GNOI().Default(t)
 	for _, tc := range cases {
 		t.Run(tc.desc, func(t *testing.T) {
+			time.Sleep(1 * time.Second) // some devices do not allow back to back traceroute to prevent flooding
+			if *deviations.TraceRouteL4ProtocolUDP {
+				if tc.defaultL4Protocol {
+					tc.traceRequest.L4Protocol = spb.TracerouteRequest_UDP
+				}
+				if tc.traceRequest.L4Protocol != spb.TracerouteRequest_UDP {
+					t.Skip("Test is skiped due to the TraceRouteL4ProtocolUDP deviatiopn")
+				}
+			}
+			if *deviations.TraceRouteFragmentation {
+				if tc.traceRequest.DoNotFragment {
+					t.Skip("Test is skiped due to the TraceRouteFragmentation deviatiopn")
+				}
+			}
 			t.Logf("Sent traceroute request: %v\n\n", tc.traceRequest)
 			traceClient, err := gnoiClient.System().Traceroute(context.Background(), tc.traceRequest)
 			if err != nil {
diff --git a/internal/deviations/deviations.go b/internal/deviations/deviations.go
index 559de90e..013d8bd5 100644
--- a/internal/deviations/deviations.go
+++ b/internal/deviations/deviations.go
@@ -127,6 +127,10 @@ var (
 
 	RoutePolicyUnderNeighborAfiSafi = flag.Bool("deviation_rpl_under_neighbor_afisafi", false, "Device requires route-policy configuration under bgp neighbor afisafi. Fully-compliant devices should pass with this deviation set to true.")
 
+	TraceRouteL4ProtocolUDP = flag.Bool("traceroute_l4_protocol_udp", false, "Device only support UDP as l4 protocol for traceroute. Use this flag to set default l4 protocol as UDP and skip the tests explictly use TCP or ICMP.")
+
+	TraceRouteFragmentation = flag.Bool("traceroute_fragmentation", false, "Device does not support fragmentation bit for traceroute.")
+
 	ConnectRetry = flag.Bool("deviation_connect_retry", false, "Connect-retry is not supported /bgp/neighbors/neighbor/timers/config/connect-retry.")
 
 	MissingBgpNeighborStatePeerGroup = flag.Bool("deviation_missing_bgp_neighbor_state_peer_group", false, "Device requires peer-group under bgp neighbor bgp/neighbors/neighbor/state/peer-group.")
