diff --git a/feature/gribi/ate_tests/weighted_balancing_test/setup_test.go b/feature/gribi/ate_tests/weighted_balancing_test/setup_test.go
index 47001584..c478b615 100644
--- a/feature/gribi/ate_tests/weighted_balancing_test/setup_test.go
+++ b/feature/gribi/ate_tests/weighted_balancing_test/setup_test.go
@@ -175,7 +175,9 @@ func configureDUT(t testing.TB, dut *ondatra.DUTDevice) {
 	// down, the device does not attempt to route packets through the
 	// default gateway 0.0.0.0/0.  Packets destined to the more specific
 	// next hop CIDRs will be routed.
-	static := &oc.NetworkInstance_Protocol_Static{
+	// XR use recursive for null route that cuases issues for trafic
+	// need to discuss with google about this
+	/*static := &oc.NetworkInstance_Protocol_Static{
 		Prefix: ygot.String(discardCIDR),
 	}
 	static.GetOrCreateNextHop("AUTO_drop_2").
@@ -184,7 +186,7 @@ func configureDUT(t testing.TB, dut *ondatra.DUTDevice) {
 		Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *deviations.StaticProtocolName).
 		Static(discardCIDR)
 	fptest.LogQuery(t, "discard route", staticp.Config(), static)
-	gnmi.Replace(t, dut, staticp.Config(), static)
+	gnmi.Replace(t, dut, staticp.Config(), static)*/
 
 	for _, dp := range dut.Ports() {
 		if i := dutInterface(dp); i != nil {
