diff --git a/feature/gribi/ate_tests/cisco/ha/ha_test.go b/feature/gribi/ate_tests/cisco/ha/ha_test.go
index 50e4c75b..cfc32bbd 100644
--- a/feature/gribi/ate_tests/cisco/ha/ha_test.go
+++ b/feature/gribi/ate_tests/cisco/ha/ha_test.go
@@ -53,7 +53,7 @@ const (
 	active_rp             = "0/RP0/CPU0"
 	standby_rp            = "0/RP1/CPU0"
 	lc                    = "0/0/CPU0" // set value for lc_oir tc, if empty it means no lc, example: 0/0/CPU0
-	with_scale            = false      // run entire script with or without scale
+	with_scale            = true       // run entire script with or without scale
 	with_RFPO             = true       // run entire script with or without RFPO
 	process_restart_count = 1
 	microdropsRepeat      = 1
