diff --git a/binding/ixweb/ixstats.go b/binding/ixweb/ixstats.go
index be5bb4a..8d47431 100644
--- a/binding/ixweb/ixstats.go
+++ b/binding/ixweb/ixstats.go
@@ -16,7 +16,6 @@ package ixweb
 
 import (
 	"bytes"
-	"golang.org/x/net/context"
 	"encoding/csv"
 	"errors"
 	"fmt"
@@ -27,7 +26,9 @@ import (
 	"sync/atomic"
 	"time"
 
-	"github.com/openconfig/gocloser"
+	"golang.org/x/net/context"
+
+	closer "github.com/openconfig/gocloser"
 )
 
 const (
@@ -37,6 +38,8 @@ const (
 	EgressStatsCaption = "EgressStatView"
 	// TrafficItemStatsCaption is the name of the trafic item statics view.
 	TrafficItemStatsCaption = "Traffic Item Statistics"
+	PortStatsCaption        = "Port Statistics"
+	FlowStatsCaption        = "Flow Statistics"
 
 	viewsPath   = "statistics/view"
 	viewOpsPath = viewsPath + "/operations"
@@ -173,7 +176,7 @@ type StatRow map[string]string
 // If drilldowns are specified, the results of each drilldown on each row of
 // the target table are also returned.
 func (v *StatView) FetchTable(ctx context.Context, drilldowns ...string) (StatTable, error) {
-	if v.caption == EgressStatsCaption {
+	if v.caption == EgressStatsCaption || v.caption == TrafficItemStatsCaption || v.caption == PortStatsCaption || v.caption == FlowStatsCaption {
 		if len(drilldowns) != 0 {
 			return nil, fmt.Errorf("drilldowns not supported for %q", EgressStatsCaption)
 		}
