diff --git a/traffic.go b/traffic.go
index 1f7e29b..9f396fb 100644
--- a/traffic.go
+++ b/traffic.go
@@ -42,7 +42,10 @@ func (tr *Traffic) String() string {
 //	frame rate: 10% of the line rate
 //	header addresses inferred: true
 func (tr *Traffic) NewFlow(name string) *Flow {
-	return &Flow{pb: &opb.Flow{Name: name}}
+	f := &Flow{pb: &opb.Flow{Name: name,}}
+	f.pb.FrameRate = &opb.FrameRate{Type: &opb.FrameRate_Fps{Fps:100}}
+	f.pb.FrameRate = &opb.FrameRate{Type: &opb.FrameRate_Bps{Bps: 1000}}
+	return f
 }
 
 // Start pushes and starts provided traffic flows on the ATE.
@@ -168,13 +171,13 @@ func (f *Flow) WithFrameRatePct(pct float64) *Flow {
 
 // WithFrameRateFPS sets the flow's rate to be n frames per second.
 func (f *Flow) WithFrameRateFPS(n uint64) *Flow {
-	f.pb.FrameRate = &opb.FrameRate{Type: &opb.FrameRate_Fps{Fps: n}}
+	f.pb.FrameRate = &opb.FrameRate{Type: &opb.FrameRate_Fps{Fps: 100}}
 	return f
 }
 
 // WithFrameRateBPS sets the flow's rate to be n bits per second.
 func (f *Flow) WithFrameRateBPS(n uint64) *Flow {
-	f.pb.FrameRate = &opb.FrameRate{Type: &opb.FrameRate_Bps{Bps: n}}
+	f.pb.FrameRate = &opb.FrameRate{Type: &opb.FrameRate_Bps{Bps: 1000}}
 	return f
 }
 
