diff --git a/config/networkinstance/telemetry-1.go b/config/networkinstance/telemetry-1.go
index cb1ab95..ac11a87 100644
--- a/config/networkinstance/telemetry-1.go
+++ b/config/networkinstance/telemetry-1.go
@@ -14053,24 +14053,32 @@ func (n *NetworkInstance_ProtocolPath) BatchDelete(t testing.TB, b *config.SetRe
 // Replace replaces the configuration at /openconfig-network-instance/network-instances/network-instance/protocols/protocol.
 func (n *NetworkInstance_ProtocolPath) Replace(t testing.TB, val *oc.NetworkInstance_Protocol) *gpb.SetResponse {
 	t.Helper()
+       name:="default"
+    	val.Name=&name
 	return genutil.Replace(t, n, val)
 }
 
 // BatchReplace buffers a config replace operation at /openconfig-network-instance/network-instances/network-instance/protocols/protocol in the given batch object.
 func (n *NetworkInstance_ProtocolPath) BatchReplace(t testing.TB, b *config.SetRequestBatch, val *oc.NetworkInstance_Protocol) {
 	t.Helper()
+	name:="default"
+   val.Name=&name
 	b.BatchReplace(t, n, val)
 }
 
 // Update updates the configuration at /openconfig-network-instance/network-instances/network-instance/protocols/protocol.
 func (n *NetworkInstance_ProtocolPath) Update(t testing.TB, val *oc.NetworkInstance_Protocol) *gpb.SetResponse {
 	t.Helper()
+       name:="default"
+    	val.Name=&name
 	return genutil.Update(t, n, val)
 }
 
 // BatchUpdate buffers a config update operation at /openconfig-network-instance/network-instances/network-instance/protocols/protocol in the given batch object.
 func (n *NetworkInstance_ProtocolPath) BatchUpdate(t testing.TB, b *config.SetRequestBatch, val *oc.NetworkInstance_Protocol) {
 	t.Helper()
+       name:="default"
+    	val.Name=&name
 	b.BatchUpdate(t, n, val)
 }
 
