diff --git a/config/networkinstance/telemetry-0.go b/config/networkinstance/telemetry-0.go
index 8df392a..0a009c8 100644
--- a/config/networkinstance/telemetry-0.go
+++ b/config/networkinstance/telemetry-0.go
@@ -85,18 +85,24 @@ func (n *NetworkInstancePath) BatchDelete(t testing.TB, b *config.SetRequestBatc
 // Replace replaces the configuration at /openconfig-network-instance/network-instances/network-instance.
 func (n *NetworkInstancePath) Replace(t testing.TB, val *oc.NetworkInstance) *gpb.SetResponse {
 	t.Helper()
+	// diable setting type for network insatnce
+   val.Type=oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_UNSET
 	return genutil.Replace(t, n, val)
 }
 
 // BatchReplace buffers a config replace operation at /openconfig-network-instance/network-instances/network-instance in the given batch object.
 func (n *NetworkInstancePath) BatchReplace(t testing.TB, b *config.SetRequestBatch, val *oc.NetworkInstance) {
 	t.Helper()
+	// diable setting type for network insatnce
+   val.Type=oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_UNSET
 	b.BatchReplace(t, n, val)
 }
 
 // Update updates the configuration at /openconfig-network-instance/network-instances/network-instance.
 func (n *NetworkInstancePath) Update(t testing.TB, val *oc.NetworkInstance) *gpb.SetResponse {
 	t.Helper()
+   // diable setting type for network insatnce
+   val.Type=oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_UNSET
 	return genutil.Update(t, n, val)
 }
 
