diff --git a/internal/junitxml/junitxml.go b/internal/junitxml/junitxml.go
index 8106415..0926081 100644
--- a/internal/junitxml/junitxml.go
+++ b/internal/junitxml/junitxml.go
@@ -16,6 +16,7 @@
 package junitxml
 
 import (
+	"bytes"
 	"encoding/xml"
 	"fmt"
 	"io"
@@ -28,8 +29,9 @@ import (
 )
 
 var (
-	convSingle *converter // Converter singleton.
-	timeNowFn  = time.Now // To be stubbed out by unit tests.
+	convSingle *converter  // Converter singleton.
+	timeNowFn  = time.Now  // To be stubbed out by unit tests.
+	oStdout    = os.Stdout // Copy of the original stdout.
 )
 
 // StartConverting starts converting a Go test log in stdout to a JUnit XML
@@ -75,7 +77,10 @@ func startConverter(dest io.Writer) (*converter, error) {
 	errCh := make(chan error)
 	conv := &converter{file: pw, errCh: errCh}
 	go func() {
-		errCh <- conv.logToXML(pr, dest)
+		var buf bytes.Buffer
+		tee := io.TeeReader(pr, &buf)
+		errCh <- conv.logToXML(tee, dest)
+		oStdout.Write(buf.Bytes())
 	}()
 	return conv, nil
 }
